

Large-Scale Metatxome Analysis: [A global ocean atlas of eukaryotic gene](https://www.nature.com/articles/s41467-017-02342-1)

Sampling Data info: [Viral to metazoan marine plankton nucleotide sequences from the Tara Oceans expedition](https://www.nature.com/articles/sdata201793)

*Depth:*

  - SRF = surface
  - DCM = deep chlorophyll maximum
  - MES = mesopelagic

*Size Fractions:*

``` "In the case of RNA issued from fractions enriched in protists and metazoans (0.8–5 μm (or 0.8–3 μm), 0.8–2,000 μm, 3–2,000 μm, 5–20 μm (or 3–20 μm), 20–180 μm and 180–2,000 μm membrane filters), methods including a poly(A)+ RNA selection step were chosen. Whereas this approach is very efficient in lowering the number of rRNA reads, it does not allow to retrotranscribe mRNAs from prokaryotic species, thus leading to eukaryote-only metatranscriptomes." ```

*Sequencing_Type:*

*TS_RNA:*   

```"When at least 2 μg total RNA were available, cDNA synthesis was carried out using the TruSeq mRNA Sample preparation kit (Illumina, San Diego, CA) (Method ID: TS_RNA). Briefly, poly(A)+ RNA was selected with oligo(dT) beads, chemically fragmented and converted into single-stranded cDNA using random hexamer priming. Then, the second strand was generated to create double-stranded cDNA. Next, library preparation was performed according to the protocol described for viral metagenomics libraries
by omitting cDNA shearing and performing a post-PCR 1x AMPure XP purification." ```

*SMART_dt:*  

```"RNA extractions yielding insufficient quantities for TruSeq preparations were processed using the SMARTer Ultra Low RNA Kit (Clontech, Mountain View, CA) (Method ID: SMART_dT). This method, successfully used for eukaryotic single cell transcriptomic studies44,45, converts poly(A)+ RNA to full-length cDNA using a modified oligo(dT) primer combined with SMART (Switching Mechanism at the 5′ end of RNA Template) technology. Fifty nanograms or less total RNA were used for cDNA synthesis,followed
by 12 cycles of PCR preamplification of cDNA. Before Illumina library preparation, 5–50 ng double stranded cDNA were fragmented to a 150–600 bp size range using the E210 Covaris instrument. Then, sheared cDNA were used for Illumina library preparation following the protocol described for viral metagenomes libraries, except for the post amplification AMPure XP purification performed at a ratio 1:1." ```

*MetaG:* 

```"As short read lengths may be challenging for de novo assemblies, library preparation protocols for complex metagenomics samples were improved in order to generate much longer reads by overlapping and merging read pairs before assembly. For this purpose, a size selection step was added at the end of library preparation obtaining narrowly sized libraries around 300 bp. This corresponded to an insert fragment at around 180 bp, allowing ~20 bp paired read overlaps." ```

