



<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
      
        <link rel="canonical" href="https://bluegenes.github.io/2018-metatranscriptomics/count_transcriptomes/">
      
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="..">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-3.0.3">
    
    
      
        <title>Count transcriptomes - 2018 CICESE Metatranscriptomics workshop</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/application.451f80e5.css">
      
        <link rel="stylesheet" href="../assets/stylesheets/application-palette.22915126.css">
      
      
        
        
        <meta name="theme-color" content="#009688">
      
    
    
      <script src="../assets/javascripts/modernizr.1aa3b519.js"></script>
    
    
      <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
      
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../assets/fonts/material-icons.css">
    
      <link rel="stylesheet" href="../css/custom.css">
    
    
  </head>
  
    
    
    <body dir="ltr" data-md-color-primary="teal" data-md-color-accent="teal">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448"
    viewBox="0 0 416 448" id="__github">
  <path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19-18.125
        8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19 18.125-8.5
        18.125 8.5 10.75 19 3.125 20.5zM320 304q0 10-3.125 20.5t-10.75
        19-18.125 8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19
        18.125-8.5 18.125 8.5 10.75 19 3.125 20.5zM360
        304q0-30-17.25-51t-46.75-21q-10.25 0-48.75 5.25-17.75 2.75-39.25
        2.75t-39.25-2.75q-38-5.25-48.75-5.25-29.5 0-46.75 21t-17.25 51q0 22 8
        38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0
        37.25-1.75t35-7.375 30.5-15 20.25-25.75 8-38.375zM416 260q0 51.75-15.25
        82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5-41.75
        1.125q-19.5 0-35.5-0.75t-36.875-3.125-38.125-7.5-34.25-12.875-30.25-20.25-21.5-28.75q-15.5-30.75-15.5-82.75
        0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25
        30.875q36.75-8.75 77.25-8.75 37 0 70 8 26.25-20.5
        46.75-30.25t47.25-9.75q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34
        99.5z" />
</svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="../#counting-the-number-of-transcriptomes-in-a-metatranscriptome" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://bluegenes.github.io/2018-metatranscriptomics" title="2018 CICESE Metatranscriptomics workshop" class="md-header-nav__button md-logo">
          
            <i class="md-icon">dns</i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            
              <span class="md-header-nav__topic">
                2018 CICESE Metatranscriptomics workshop
              </span>
              <span class="md-header-nav__topic">
                Count transcriptomes
              </span>
            
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  


  <a href="https://github.com/bluegenes/2018-metatranscriptomics-workshop-dev" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#__github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      2018-metatranscriptomics-workshop-dev
    </div>
  </a>

          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="https://bluegenes.github.io/2018-metatranscriptomics" title="2018 CICESE Metatranscriptomics workshop" class="md-nav__button md-logo">
      
        <i class="md-icon">dns</i>
      
    </a>
    2018 CICESE Metatranscriptomics workshop
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/bluegenes/2018-metatranscriptomics-workshop-dev" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#__github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      2018-metatranscriptomics-workshop-dev
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href=".." title="Index" class="md-nav__link">
      Index
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../working-with-bioconda/" title="Bioconda" class="md-nav__link">
      Bioconda
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#counting-the-number-of-transcriptomes-in-a-metatranscriptome" title="Counting the number of transcriptomes in a metatranscriptome" class="md-nav__link">
    Counting the number of transcriptomes in a metatranscriptome
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#other-proteins-to-use-for-quantifcation" title="Other proteins to use for quantifcation" class="md-nav__link">
    Other proteins to use for quantifcation
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                  <h1>Count transcriptomes</h1>
                
                <h2 id="counting-the-number-of-transcriptomes-in-a-metatranscriptome">Counting the number of transcriptomes in a metatranscriptome</h2>
<p>In metagenomics, we use binning to approximate the number of genomes (or species) in a sample.
Although we can apply taxonomic classification methods like sourmash to metatranscriptomic 
samples, these methods only work reliably when sequences already exist for the organism of 
interest. This is still a relatively rare occurrence in environmental metatranscriptomics. </p>
<p>We will make use of other data in our samples to approximate the number of species present.
Although our samples were poly-A selected, inevitably a some ribosomal RNA is always sequenced
given its abundance in samples. We can estimate the number of species in our metatranscriptome
by counting the unique number of ribosomal protein sequences we see. There are many ribosomal
proteins one could use for this analysis, and we provide a table of these at the bottom of 
this lesson. We will use three sequences here, and average the number of unique sequences we 
detect between the three sequences to estimate the number of transcriptomes in our sample.</p>
<p>We will first search for these proteins in the amino acid sequences 
derived from our de novo assembly of our metatranscriptome. We will use
<a href="https://pfam.xfam.org/">Pfam</a> domains and a tool called <a href="http://hmmer.org/">HMMER</a> 
to help us locate all of the matching sequences. Then, we will XXX. Because we are only searching in our assembly, this method only captures the number of transcriptomes that assembled.</p>
<h2 id="_1"></h2>
<p>First let's download some Pfam domains. </p>
<pre><code>wget
</code></pre>

<p>Next we'll build a HMM profile of the Pfam domains. </p>
<pre><code>hmmbuild
</code></pre>

<p>We then use the HMM profile to search the proteins from our assembly</p>
<pre><code>hmmscan
</code></pre>

<p>Let's take a look at one of the files output by this search</p>
<pre><code>less -S
</code></pre>

<p>The xx column contains the names of our protein sequences that matched
these domains. We can use those names to extract our matches from 
our assembly</p>
<pre><code># Grab the names

# extract the matches
</code></pre>

<p>Let's count the number of sequences that matched </p>
<pre><code>grep &quot;&gt;&quot; XXX | wc -l
</code></pre>

<p>Some matches are quite similar to each other. Let's cluster our sequences
at 97% similarity and see how this changes the number of unique proteins
we detect. </p>
<pre><code>cdhit ...
</code></pre>

<p>From this, we estimate there are XX transcriptomes assembled from our
metatranscriptome.</p>
<p>This technique can also be used to find any Pfam domain of interest. 
For instance, if you are interested in photosynthesis, you could
extract all photosynthetic proteins by searching with the Pfam domains.</p>
<h2 id="other-proteins-to-use-for-quantifcation">Other proteins to use for quantifcation</h2>
<p>We used three proteins to quantify the number of transcriptomes in our
sample. There are more that can be used, and they are list in the table
below. They are originally derived from <a href="https://www.nature.com/articles/s41467-017-02342-1#Sec19">Carradec et al. 2018</a>.</p>
<table>
<thead>
<tr>
<th>name</th>
<th>COG</th>
<th>PFAM</th>
</tr>
</thead>
<tbody>
<tr>
<td>RpsG</td>
<td>COG0049</td>
<td>PF00177</td>
</tr>
<tr>
<td>RpsB</td>
<td>COG0052</td>
<td>PF00318</td>
</tr>
<tr>
<td>RplK</td>
<td>COG0080</td>
<td>PF03946</td>
</tr>
<tr>
<td>RplA</td>
<td>COG0081</td>
<td>PF00687</td>
</tr>
<tr>
<td>RplC</td>
<td>COG0087</td>
<td>PF00297</td>
</tr>
<tr>
<td>RplD</td>
<td>COG0088</td>
<td>PF00573</td>
</tr>
<tr>
<td>RplV</td>
<td>COG0091</td>
<td>PF00237</td>
</tr>
<tr>
<td>RpsC</td>
<td>COG0092</td>
<td>PF00189</td>
</tr>
<tr>
<td>RplN</td>
<td>COG0093</td>
<td>PF01929</td>
</tr>
<tr>
<td>RplE</td>
<td>COG0094</td>
<td>PF00281</td>
</tr>
<tr>
<td>RpsH</td>
<td>COG0096</td>
<td>PF00410</td>
</tr>
<tr>
<td>RplF</td>
<td>COG0097</td>
<td>NA</td>
</tr>
<tr>
<td>RpsE</td>
<td>COG0098</td>
<td>PF03719, PF00333</td>
</tr>
<tr>
<td>RpsM</td>
<td>COG0099</td>
<td>PF00416</td>
</tr>
<tr>
<td>RpsK</td>
<td>COG0100</td>
<td>PF00411</td>
</tr>
<tr>
<td>RplM</td>
<td>COG0102</td>
<td>PF00572</td>
</tr>
<tr>
<td>RpsI</td>
<td>COG0103</td>
<td>PF00380</td>
</tr>
<tr>
<td>RpsO</td>
<td>COG0184</td>
<td>NA</td>
</tr>
<tr>
<td>RpsS</td>
<td>COG0185</td>
<td>PF00203</td>
</tr>
<tr>
<td>RpsQ</td>
<td>COG0186</td>
<td>PF00366</td>
</tr>
<tr>
<td>GyrB</td>
<td>COG0187</td>
<td>PF00204</td>
</tr>
<tr>
<td>GyrA</td>
<td>COG0188</td>
<td>PF00521</td>
</tr>
<tr>
<td>RimK</td>
<td>COG0189</td>
<td>PF08443</td>
</tr>
<tr>
<td>FolD</td>
<td>COG0190</td>
<td>PF00763, PF02882</td>
</tr>
<tr>
<td>Fba</td>
<td>COG0191</td>
<td>NA</td>
</tr>
<tr>
<td>MetK</td>
<td>COG0192</td>
<td>PF01941</td>
</tr>
<tr>
<td>Pth</td>
<td>COG0193</td>
<td>PF01195</td>
</tr>
<tr>
<td>Gmk</td>
<td>COG0194</td>
<td>PF00625</td>
</tr>
<tr>
<td>RplO</td>
<td>COG0200</td>
<td>PF00827</td>
</tr>
<tr>
<td>RplR</td>
<td>COG0256</td>
<td>NA</td>
</tr>
<tr>
<td>RpsD</td>
<td>COG0522</td>
<td>PF00163</td>
</tr>
</tbody>
</table>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright &copy; 2018 <a href="http://ivory.idyll.org/lab/">Lab for Data Intensive Biology</a> at UC Davis
          </div>
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/application.e72fd936.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:".."}})</script>
      
        <script src="../search/main.js"></script>
      
    
    
      
    
  </body>
</html>