



<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
      
        <link rel="canonical" href="https://ngs-docs.github.io/2018-cicese-metatranscriptomics/short-read-quality-control/">
      
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching pages">
      
        <meta name="lang:search.result.one" content="1 matching page">
      
        <meta name="lang:search.result.other" content="# matching page">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="..">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-3.0.3">
    
    
      
        <title>Short Read Quality Control - 2018 CICESE Metatranscriptomics workshop</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/application.451f80e5.css">
      
        <link rel="stylesheet" href="../assets/stylesheets/application-palette.22915126.css">
      
      
        
        
        <meta name="theme-color" content="#009688">
      
    
    
      <script src="../assets/javascripts/modernizr.1aa3b519.js"></script>
    
    
      <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
      
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../assets/fonts/material-icons.css">
    
      <link rel="stylesheet" href="../css/custom.css">
    
    
  </head>
  
    
    
    <body dir="ltr" data-md-color-primary="teal" data-md-color-accent="teal">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448"
    viewBox="0 0 416 448" id="__github">
  <path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19-18.125
        8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19 18.125-8.5
        18.125 8.5 10.75 19 3.125 20.5zM320 304q0 10-3.125 20.5t-10.75
        19-18.125 8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19
        18.125-8.5 18.125 8.5 10.75 19 3.125 20.5zM360
        304q0-30-17.25-51t-46.75-21q-10.25 0-48.75 5.25-17.75 2.75-39.25
        2.75t-39.25-2.75q-38-5.25-48.75-5.25-29.5 0-46.75 21t-17.25 51q0 22 8
        38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0
        37.25-1.75t35-7.375 30.5-15 20.25-25.75 8-38.375zM416 260q0 51.75-15.25
        82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5-41.75
        1.125q-19.5 0-35.5-0.75t-36.875-3.125-38.125-7.5-34.25-12.875-30.25-20.25-21.5-28.75q-15.5-30.75-15.5-82.75
        0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25
        30.875q36.75-8.75 77.25-8.75 37 0 70 8 26.25-20.5
        46.75-30.25t47.25-9.75q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34
        99.5z" />
</svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="../#short-read-quality-control" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://ngs-docs.github.io/2018-cicese-metatranscriptomics" title="2018 CICESE Metatranscriptomics workshop" class="md-header-nav__button md-logo">
          
            <i class="md-icon">dns</i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            
              <span class="md-header-nav__topic">
                2018 CICESE Metatranscriptomics workshop
              </span>
              <span class="md-header-nav__topic">
                Short Read Quality Control
              </span>
            
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search this site" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching this site
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  


  <a href="https://github.com/ngs-docs/2018-cicese-metatranscriptomics" title="Go to repository" class="md-source" data-md-source="github">
    
    <div class="md-source__repository">
      2018-cicese-metatranscriptomics
    </div>
  </a>

          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="https://ngs-docs.github.io/2018-cicese-metatranscriptomics" title="2018 CICESE Metatranscriptomics workshop" class="md-nav__button md-logo">
      
        <i class="md-icon">dns</i>
      
    </a>
    2018 CICESE Metatranscriptomics workshop
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/ngs-docs/2018-cicese-metatranscriptomics" title="Go to repository" class="md-source" data-md-source="github">
    
    <div class="md-source__repository">
      2018-cicese-metatranscriptomics
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-1" type="checkbox" id="nav-1">
    
    <label class="md-nav__link" for="nav-1">
      Overview
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-1">
        Overview
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href=".." title="Schedule" class="md-nav__link">
      Schedule
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../code-of-conduct/" title="Code of Conduct" class="md-nav__link">
      Code of Conduct
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2" checked>
    
    <label class="md-nav__link" for="nav-2">
      Day 1
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Day 1
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../welcome/" title="Bienvenidos" class="md-nav__link">
      Bienvenidos
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../cicese-cluster/" title="Cluster login" class="md-nav__link">
      Cluster login
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../setting-up-tara-environment/" title="Software setup" class="md-nav__link">
      Software setup
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../tara-sample-data/" title="TARA Sample Data" class="md-nav__link">
      TARA Sample Data
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        Short Read Quality Control
      </label>
    
    <a href="./" title="Short Read Quality Control" class="md-nav__link md-nav__link--active">
      Short Read Quality Control
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#data-source" title="Data source" class="md-nav__link">
    Data source
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#set-up-workspace-and-download-the-data" title="Set up workspace and download the data" class="md-nav__link">
    Set up workspace and download the data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#quality-trimming-and-light-quality-filtering" title="Quality trimming and light quality filtering" class="md-nav__link">
    Quality trimming and light quality filtering
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#link-your-data-into-your-working-directory" title="Link your data into your working directory" class="md-nav__link">
    Link your data into your working directory
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fastqc" title="FastQC" class="md-nav__link">
    FastQC
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#multiqc" title="MultiQC" class="md-nav__link">
    MultiQC
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#adapter-trim-each-pair-of-files" title="Adapter trim each pair of files" class="md-nav__link">
    Adapter trim each pair of files
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../sourmash-taxonomic-classification/" title="Sourmash Taxonomic Classification" class="md-nav__link">
      Sourmash Taxonomic Classification
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../sample-comparison/" title="Comparing Samples" class="md-nav__link">
      Comparing Samples
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../khmer-trimming/" title="Khmer Error Trimming" class="md-nav__link">
      Khmer Error Trimming
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../megahit-assembly/" title="Assembly with Megahit" class="md-nav__link">
      Assembly with Megahit
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../count_transcriptomes/" title="Count Transcriptomes" class="md-nav__link">
      Count Transcriptomes
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3">
    
    <label class="md-nav__link" for="nav-3">
      Day 2
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        Day 2
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../evaluation/" title="Assembly Evaluation" class="md-nav__link">
      Assembly Evaluation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../read-mapping/" title="Mapping reads to the Assembly" class="md-nav__link">
      Mapping reads to the Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../annotation/" title="Assembly Annotation" class="md-nav__link">
      Assembly Annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../anvio/" title="Visualize with anvio" class="md-nav__link">
      Visualize with anvio
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../working-with-bioconda/" title="Bioconda" class="md-nav__link">
      Bioconda
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#data-source" title="Data source" class="md-nav__link">
    Data source
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#set-up-workspace-and-download-the-data" title="Set up workspace and download the data" class="md-nav__link">
    Set up workspace and download the data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#quality-trimming-and-light-quality-filtering" title="Quality trimming and light quality filtering" class="md-nav__link">
    Quality trimming and light quality filtering
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#link-your-data-into-your-working-directory" title="Link your data into your working directory" class="md-nav__link">
    Link your data into your working directory
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fastqc" title="FastQC" class="md-nav__link">
    FastQC
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#multiqc" title="MultiQC" class="md-nav__link">
    MultiQC
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#adapter-trim-each-pair-of-files" title="Adapter trim each pair of files" class="md-nav__link">
    Adapter trim each pair of files
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                <h1 id="short-read-quality-control">Short Read Quality Control</h1>
<p>You should now be logged into your cluster! If not, go back 
to <a href="../cicese-cluster/">Cluster Login</a></p>
<h2 id="data-source">Data source</h2>
<p>We will be using RNAseq reads from a small subset of data from the <a href="https://oceans.taraexpeditions.org/en/m/about-tara/les-expeditions/tara-oceans">TARA Oceans Expedition</a>, 
from <a href="https://www.nature.com/articles/sdata201793#t1">Alberti et al., 2017</a> and analyzed as part of <a href="https://www.nature.com/articles/s41467-017-02342-1">A global ocean atlas of eukaryotic genes</a>.</p>
<h2 id="set-up-workspace-and-download-the-data">Set up workspace and download the data</h2>
<p>First, make some directories to work in:</p>
<pre><code>cd
mkdir -p work/data
</code></pre>

<p>Next, change into the data dir and download the data subsets:</p>
<pre><code>cd work/data
wget https://osf.io/76qm3/download -O tara135_1m.zip
wget https://osf.io/y5dfh/download -O tara136-137_1m.zip
</code></pre>

<p>Now, let's unzip and make the files difficult to delete</p>
<pre><code>unzip tara135_1m.zip
unzip tara136-137_1m.zip

chmod u-w *fq.gz
</code></pre>

<p>To make life easier, let's define a variable for the location
of this tara working directory:</p>
<pre><code> export PROJECT=~/work
</code></pre>

<p>Check that your data is where it should be</p>
<pre><code>ls $PROJECT/data/
</code></pre>

<p>If you see all the files you think you should, good!  Otherwise, debug.</p>
<p>These are FASTQ files -- let's take a look at them:</p>
<pre><code>zless $PROJECT/data//TARA_135_DCM_5-20_rep1_1m_1.fq.gz
</code></pre>

<p>(use the spacebar to scroll down, and type 'q' to exit 'zless')</p>
<p>Question:</p>
<ul>
<li>where does the filename come from?</li>
<li>why are there 1 and 2 in the file names?</li>
</ul>
<p>Links:</p>
<ul>
<li><a href="http://en.wikipedia.org/wiki/FASTQ_format">FASTQ Format</a></li>
</ul>
<h2 id="quality-trimming-and-light-quality-filtering">Quality trimming and light quality filtering</h2>
<p>Make sure you've got the PROJECT location defined, and your data is there:</p>
<pre><code>set -u
printf &quot;\nMy raw data is in $PROJECT/data/, and consists of $(ls -1 ${PROJECT}/data/*.fq.gz | wc -l) files\n\n&quot;
set +u
</code></pre>

<p><em>Important:</em> If you get an error above or the count of files is wrong...STOP!! Revisit the download &amp; unzip instructions!</p>
<h3 id="link-your-data-into-your-working-directory">Link your data into your working directory</h3>
<p>Change into your project directory and make a workspace for quality trimming:</p>
<pre><code>cd ${PROJECT}
mkdir -p quality
cd quality
</code></pre>

<p>Now, link the data files into your new workspace</p>
<pre><code>ln -s ../data/*.fq.gz ./
</code></pre>

<p>(Linking with <code>ln</code> avoids having to make a copy of the files, which will take up storage space.)</p>
<p>Check to make sure it worked</p>
<pre><code>printf &quot;I see $(ls -1 *.fq.gz | wc -l) files here.\n&quot;
</code></pre>

<p>You can also do an <code>ls</code> to list the files.</p>
<p>If you see only one entry, <code>*.fq.gz</code>, then the ln command above didn't work properly.  One possibility is that your files aren't in your data directory; another is that their names don't end with
<code>.fq.gz</code>.</p>
<h3 id="fastqc">FastQC</h3>
<p>We're going to use
<a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> to summarize the data. </p>
<p>To install fastqc via conda:
Note: make sure you've followed the <a href="../working-with-bioconda/">conda setup instructions</a>.</p>
<pre><code>conda install fastqc
</code></pre>

<p>Now, run FastQC on the files:</p>
<pre><code>fastqc *.fq.gz
</code></pre>

<p>After this finishes running (has to run on each file so might take a while), type 'ls':</p>
<pre><code>ls -d *fastqc.zip*
</code></pre>

<p>to list the files, and you should see a number of files with the extensions <code>.fastqc.zip</code>.</p>
<p>Inside each of the fastqc directories you will find reports from the fastqc. You can download these files using a technique we'll show you later.</p>
<p>Questions:</p>
<ul>
<li>What should you pay attention to in the FastQC report?</li>
<li>Which is "better", file 1 or file 2? And why?</li>
</ul>
<p>Links:</p>
<ul>
<li><a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a></li>
<li><a href="http://www.youtube.com/watch?v=bz93ReOv87Y">FastQC tutorial video</a></li>
<li><a href="http://angus.readthedocs.io/en/2015/_static/2015-lecture2-sequencing.pptx.pdf">Examples of fastqc after technical sequencer problems</a>(starting on slide 40)</li>
</ul>
<p>There are several caveats about FastQC - the main one is that it only calculates certain statistics (like duplicated sequences) for subsets
of the data (e.g. duplicate sequences are only analyzed for the first 100,000 sequences in each file.</p>
<h3 id="multiqc">MultiQC</h3>
<p>If you would like to aggregate all of your fastqc reports across many samples, <a href="http://multiqc.info/">MultiQC</a> will do this into a single report for easy comparison.</p>
<p>Install MultiQC with conda:</p>
<pre><code>conda install multiqc
</code></pre>

<p>Run MultiQC:</p>
<pre><code>multiqc .
</code></pre>

<p>The terminal output should look like this:</p>
<pre><code>[INFO   ]         multiqc : This is MultiQC v1.6
[INFO   ]         multiqc : Template    : default
[INFO   ]         multiqc : Searching '.'
[INFO   ]          fastqc : Found 8 reports
[INFO   ]         multiqc : Compressing plot data
[INFO   ]         multiqc : Report      : multiqc_report.html
[INFO   ]         multiqc : Data        : multiqc_data
[INFO   ]         multiqc : MultiQC complete
</code></pre>

<p>You can view this report <a href="../files/multiqc_report.html">here</a></p>
<p>If you are unable to use scp though a terminal output, you can see the fastqc html output <a href="../files/TARA_135_DCM_5-20_rep1_1m_1_fastqc.html">here</a></p>
<h3 id="adapter-trim-each-pair-of-files">Adapter trim each pair of files</h3>
<p>Install Trimmomatic:</p>
<pre><code>conda install trimmomatic
</code></pre>

<p>Setup trim directory:</p>
<pre><code>cd $PROJECT
mkdir -p trim
cd trim
ln -s ../data/*.fq.gz .
cat /LUSTRE/apps/workshop/miniconda3/envs/tara/share/trimmomatic*/adapters/* &gt; combined.fa
</code></pre>

<p>See this excellent paper on trimming from <a href="http://journal.frontiersin.org/article/10.3389/fgene.2014.00013/full">MacManes 2014</a>.</p>
<p>Run:</p>
<pre><code>for filename in *1.fq.gz
do
#Use the program basename to remove _1.fq.gz to generate the base
base=$(basename $filename _1.fq.gz)
echo $base

# run Trimmomatic
trimmomatic PE ${base}_1.fq.gz \
              ${base}_2.fq.gz \
     ${base}_1.qc.fq.gz s1_se \
     ${base}_2.qc.fq.gz s2_se \
     ILLUMINACLIP:combined.fa:2:40:15 \
     LEADING:2 TRAILING:2 \
     SLIDINGWINDOW:4:2 \
     MINLEN:25
# save the orphans
gzip -9c s1_se s2_se &gt;&gt; orphans.qc.fq.gz
rm -f s1_se s2_se

done
</code></pre>

<p>Now, run fastqc again on trimmed files:</p>
<pre><code>fastqc *.qc.fq.gz
multiqc .
</code></pre>

<p>The paired sequences output by this set of commands will be in the files ending in <code>.qc.fq.gz</code>, with any orphaned sequences all together
in <code>orphans.qc.fq.gz</code>.</p>
<p>Make these trimmed reads read-only and keep them, as we will reuse them later.</p>
<pre><code>chmod a-w ${PROJECT}/trim/*.qc.fq.gz
</code></pre>

<p>Questions:</p>
<ul>
<li>How do you figure out what the parameters mean?</li>
<li>How do you figure out what parameters to use?</li>
<li>What adapters do you use?</li>
<li>What version of Trimmomatic are we using here? (And FastQC?)</li>
<li>Do you think parameters are different for RNAseq and genomic data sets?</li>
<li>What's with these annoyingly long and complicated filenames?</li>
<li>why are we running R1 and R2 together?</li>
</ul>
<p>For a discussion of optimal trimming strategies, see
<a href="http://journal.frontiersin.org/Journal/10.3389/fgene.2014.00013/abstract">MacManes, 2014</a>
-- it's about RNAseq but similar arguments should apply to metagenome
assembly.</p>
<p>Links:</p>
<ul>
<li><a href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic</a></li>
</ul>
<p>Questions:</p>
<ul>
<li>is the quality trimmed data "better" than before?</li>
<li>Does it matter that you still have adapters!?</li>
</ul>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../tara-sample-data/" title="TARA Sample Data" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                TARA Sample Data
              </span>
            </div>
          </a>
        
        
          <a href="../sourmash-taxonomic-classification/" title="Sourmash Taxonomic Classification" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Sourmash Taxonomic Classification
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright &copy; 2018 <a href="http://ivory.idyll.org/lab/">Lab for Data Intensive Biology</a> at UC Davis
          </div>
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/application.e72fd936.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:".."}})</script>
      
        <script src="../search/main.js"></script>
      
    
    
      
    
  </body>
</html>