



<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
      
        <link rel="canonical" href="https://bluegenes.github.io/2018-metatranscriptomics/short-read-quality-control/">
      
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="..">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-3.0.3">
    
    
      
        <title>Read QC - 2018 CICESE Metatranscriptomics workshop</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/application.451f80e5.css">
      
        <link rel="stylesheet" href="../assets/stylesheets/application-palette.22915126.css">
      
      
        
        
        <meta name="theme-color" content="#009688">
      
    
    
      <script src="../assets/javascripts/modernizr.1aa3b519.js"></script>
    
    
      <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
      
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../assets/fonts/material-icons.css">
    
      <link rel="stylesheet" href="../css/custom.css">
    
    
  </head>
  
    
    
    <body dir="ltr" data-md-color-primary="teal" data-md-color-accent="teal">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448"
    viewBox="0 0 416 448" id="__github">
  <path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19-18.125
        8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19 18.125-8.5
        18.125 8.5 10.75 19 3.125 20.5zM320 304q0 10-3.125 20.5t-10.75
        19-18.125 8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19
        18.125-8.5 18.125 8.5 10.75 19 3.125 20.5zM360
        304q0-30-17.25-51t-46.75-21q-10.25 0-48.75 5.25-17.75 2.75-39.25
        2.75t-39.25-2.75q-38-5.25-48.75-5.25-29.5 0-46.75 21t-17.25 51q0 22 8
        38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0
        37.25-1.75t35-7.375 30.5-15 20.25-25.75 8-38.375zM416 260q0 51.75-15.25
        82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5-41.75
        1.125q-19.5 0-35.5-0.75t-36.875-3.125-38.125-7.5-34.25-12.875-30.25-20.25-21.5-28.75q-15.5-30.75-15.5-82.75
        0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25
        30.875q36.75-8.75 77.25-8.75 37 0 70 8 26.25-20.5
        46.75-30.25t47.25-9.75q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34
        99.5z" />
</svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="../#short-read-quality-control" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://bluegenes.github.io/2018-metatranscriptomics" title="2018 CICESE Metatranscriptomics workshop" class="md-header-nav__button md-logo">
          
            <i class="md-icon">dns</i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            
              <span class="md-header-nav__topic">
                2018 CICESE Metatranscriptomics workshop
              </span>
              <span class="md-header-nav__topic">
                Read QC
              </span>
            
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  


  <a href="https://github.com/bluegenes/2018-metatranscriptomics-workshop-dev" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#__github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      2018-metatranscriptomics-workshop-dev
    </div>
  </a>

          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="https://bluegenes.github.io/2018-metatranscriptomics" title="2018 CICESE Metatranscriptomics workshop" class="md-nav__button md-logo">
      
        <i class="md-icon">dns</i>
      
    </a>
    2018 CICESE Metatranscriptomics workshop
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/bluegenes/2018-metatranscriptomics-workshop-dev" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#__github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      2018-metatranscriptomics-workshop-dev
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-1" type="checkbox" id="nav-1">
    
    <label class="md-nav__link" for="nav-1">
      Overview
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-1">
        Overview
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href=".." title="Schedule" class="md-nav__link">
      Schedule
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../code-of-conduct/" title="Code of Conduct" class="md-nav__link">
      Code of Conduct
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2" checked>
    
    <label class="md-nav__link" for="nav-2">
      Day 1
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Day 1
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../cicese-cluster/" title="Cluster login" class="md-nav__link">
      Cluster login
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../working-with-bioconda/" title="Bioconda" class="md-nav__link">
      Bioconda
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../tara-sample-data/" title="TARA Sample Data" class="md-nav__link">
      TARA Sample Data
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        Read QC
      </label>
    
    <a href="./" title="Read QC" class="md-nav__link md-nav__link--active">
      Read QC
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#data-source" title="Data source" class="md-nav__link">
    Data source
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#set-up-workspace-and-download-the-data" title="Set up workspace and download the data" class="md-nav__link">
    Set up workspace and download the data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#quality-trimming-and-light-quality-filtering" title="Quality trimming and light quality filtering" class="md-nav__link">
    Quality trimming and light quality filtering
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../metatranscriptome-assembly/" title="Assembly" class="md-nav__link">
      Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../metatranscriptome-evaluation/" title="Evaluation" class="md-nav__link">
      Evaluation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../count_transcriptomes/" title="Count Transcriptomes" class="md-nav__link">
      Count Transcriptomes
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3">
    
    <label class="md-nav__link" for="nav-3">
      Day 2
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        Day 2
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../metatranscriptome-annotation/" title="Annotation" class="md-nav__link">
      Annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../read-mapping/" title="Mapping" class="md-nav__link">
      Mapping
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../sourmash-taxonomic-classification/" title="Taxonomic Classification" class="md-nav__link">
      Taxonomic Classification
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../anvio/" title="Visualize with anvio" class="md-nav__link">
      Visualize with anvio
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#data-source" title="Data source" class="md-nav__link">
    Data source
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#set-up-workspace-and-download-the-data" title="Set up workspace and download the data" class="md-nav__link">
    Set up workspace and download the data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#quality-trimming-and-light-quality-filtering" title="Quality trimming and light quality filtering" class="md-nav__link">
    Quality trimming and light quality filtering
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                <h1 id="short-read-quality-control">Short Read Quality Control</h1>
<p><em>lesson changes in progress</em>
todo:
<em> change download links
</em> change names in all downstream steps
* add back in multiqc</p>
<p>You should now be logged into your cluster! If not, go back 
to <a href="../cicese-cluster/">Cluster Login</a></p>
<h2 id="data-source">Data source</h2>
<p>We will be using RNAseq reads from a small subset of data from the <a href="https://oceans.taraexpeditions.org/en/m/about-tara/les-expeditions/tara-oceans">TARA Oceans Expedition</a> 
and analyzed as part of <a href="https://www.nature.com/articles/s41467-017-02342-1">A global ocean atlas of eukaryotic genes</a>.</p>
<h2 id="set-up-workspace-and-download-the-data">Set up workspace and download the data</h2>
<p>First, make some directories to work in:</p>
<pre><code>cd
mdkir -p tara/data
</code></pre>

<p>Next, change into the data dir and download the data subsets:</p>
<pre><code>cd tara/data
#curl -L https://osf.io/365fg/download -o tara135_5-20_1m.zip
#curl -L https://osf.io/9tf2g/download -o tara136-137_5-20_1m.zip 
#unzip tara135_5-20_1m.zip
#unzip tara136-137_5-20_1m.zip
</code></pre>

<p>To make our lives easier, define a variable for the location
of this tara working directory:</p>
<pre><code> export PROJECT= ~/tara
</code></pre>

<p>Check that your data is where it should be</p>
<pre><code>ls $PROJECT/data
</code></pre>

<p>If you see all the files you think you should, good!  Otherwise, debug.</p>
<p>These are FASTQ files -- let's take a look at them:</p>
<pre><code>#less 0Hour_ATCACG_L002_R1_001.fastq
</code></pre>

<p>(use the spacebar to scroll down, and type 'q' to exit 'less')</p>
<p>Question:</p>
<ul>
<li>where does the filename come from?</li>
<li>why are there 1 and 2 in the file names?</li>
</ul>
<p>Links:</p>
<ul>
<li><a href="http://en.wikipedia.org/wiki/FASTQ_format">FASTQ Format</a></li>
</ul>
<h2 id="quality-trimming-and-light-quality-filtering">Quality trimming and light quality filtering</h2>
<p>Make sure you've got the PROJECT location defined, and your data is there:</p>
<pre><code>set -u
printf &quot;\nMy raw data is in $PROJECT/data/, and consists of $(ls -1 ${PROJECT}/data/*.fastq | wc -l) files\n\n&quot;
set +u
</code></pre>

<p><em>Important:</em> If you get an error above or the count of files is wrong...STOP!! Revisit the installation instructions!</p>
<h1 id="from-seatac-lesson">FROM SEATAC LESSON:</h1>
<h3 id="link-your-data-into-your-working-directory">Link your data into your working directory</h3>
<p>Change into your project directory and make a workspace for quality trimming:</p>
<pre><code>cd ${PROJECT}
mkdir -p quality
cd quality
</code></pre>

<p>Now, link the data files into your new workspace</p>
<pre><code>ln -s ../data/*.fastq .
</code></pre>

<p>(Linking with <code>ln</code> avoids having to make a copy of the files, which will take up storage space.)</p>
<p>Check to make sure it worked</p>
<pre><code>printf &quot;I see $(ls -1 *.fastq | wc -l) files here.\n&quot;
</code></pre>

<p>You can also do an <code>ls</code> to list the files.</p>
<p>If you see only one entry, <code>*.fastq</code>, then the ln command above didn't work properly.  One possibility is that your files aren't in your data directory; another is that their names don't end with
<code>.fastq</code>.</p>
<h3 id="fastqc">FastQC</h3>
<p>We're going to use
<a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a>
summarize the data. We already installed 'fastqc' on our computer for
you.</p>
<p>Now, run FastQC on two files:</p>
<pre><code>fastqc *.fastq
</code></pre>

<p>After this finishes running (has to run on each file so might take a while), type 'ls':</p>
<pre><code>ls -d *fastqc.zip*
</code></pre>

<p>to list the files, and you should see a number of files with the extensions <code>.fastqc.zip</code>.</p>
<p>Inside each of the fatqc directories you will find reports from the fastqc. You can download these files using your RStudio Server console, if you like. To install and run an RStudio Server, go <a href="https://angus.readthedocs.io/en/2017/visualizing-blast-scores-with-RStudio.html#installing-and-running-rstudio-on-jetstream">here</a>.</p>
<p>Questions:</p>
<ul>
<li>What should you pay attention to in the FastQC report?</li>
<li>Which is "better", file 1 or file 2? And why?</li>
</ul>
<p>Links:</p>
<ul>
<li><a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a></li>
<li><a href="http://www.youtube.com/watch?v=bz93ReOv87Y">FastQC tutorial video</a></li>
<li><a href="http://angus.readthedocs.io/en/2015/_static/2015-lecture2-sequencing.pptx.pdf">Examples of fastqc after technical sequencer problems</a>(starting on slide 40)</li>
</ul>
<p>There are several caveats about FastQC - the main one is that it only calculates certain statistics (like duplicated sequences) for subsets
of the data (e.g. duplicate sequences are only analyzed for the first 100,000 sequences in each file.</p>
<h3 id="multiqc">MultiQC</h3>
<p>If you would like to aggregate all of your fastqc reports across many samples, <a href="http://multiqc.info/">MultiQC</a> will do this into a single report for easy comparison.</p>
<p>Run MultiQC:</p>
<pre><code>multiqc .
</code></pre>

<p>The terminal output should look like this:</p>
<pre><code>[INFO   ]         multiqc : This is MultiQC v1.6
[INFO   ]         multiqc : Template    : default
[INFO   ]         multiqc : Searching '.'
[INFO   ]          fastqc : Found 20 reports
[INFO   ]         multiqc : Compressing plot data
[INFO   ]         multiqc : Report      : multiqc_report.html
[INFO   ]         multiqc : Data        : multiqc_data
[INFO   ]         multiqc : MultiQC complete
</code></pre>

<p>You can also view this output <a href="_static/multiqc_report.html">here</a></p>
<h4 id="view-your-files-on-your-own-computer">View your files on your own computer</h4>
<p>As an alternative to viewing the files on the Rstudio server, we can secure copy (scp) these files to our own laptops, and view them from there.</p>
<pre><code>mkdir ~/Desktop/nema_fastqc  # make a directory for these files
scp username@ip.address:/work/quality/*html ~/Desktop/nema_fastqc
</code></pre>

<p>where the first argument after <code>scp</code> is your login and path for the files we want to copy (from the jetstream instance), and the second argument is the path to place the files on our own computer.</p>
<p>If you are unable to use scp though a terminal output, you can see the fastqc html output <a href="_static/6Hour_CGATGT_L002_R2_003.extract_fastqc.html">here</a></p>
<h3 id="adapter-trim-each-pair-of-files">Adapter trim each pair of files</h3>
<p>Setup trim directory:</p>
<pre><code>cd ..
mkdir trim
cd trim
ln -s ../data/*.fastq .
cat /opt/miniconda3/share/trimmomatic*/adapters/* &gt; combined.fa
</code></pre>

<p>See excellent paper on trimming from <a href="http://journal.frontiersin.org/article/10.3389/fgene.2014.00013/full">MacManes 2014</a>.</p>
<p>Run:</p>
<pre><code>for filename in *_R1_*.fastq
do
# first, make the base by removing fastq
  base=$(basename $filename .fastq)
  echo $base

# now, construct the R2 filename by replacing R1 with R2
  baseR2=${base/_R1_/_R2_}
  echo $baseR2

# finally, run Trimmomatic
  trimmomatic PE ${base}.fastq ${baseR2}.fastq \
    ${base}.qc.fq.gz s1_se.gz \
    ${baseR2}.qc.fq.gz s2_se.gz \
    ILLUMINACLIP:combined.fa:2:40:15 \
    LEADING:2 TRAILING:2 \
    SLIDINGWINDOW:4:2 \
    MINLEN:25

# save the orphans
  zcat s1_se.gz s2_se.gz &gt;&gt; orphans.qc.fq.gz
  rm -f s1_se.gz s2_se.gz
done

</code></pre>

<p>Now, run fastqc again on trimmed files:</p>
<pre><code>fastqc *.qc.fq.gz
multiqc .
</code></pre>

<p>The paired sequences output by this set of commands will be in the files ending in <code>.qc.fq.gz</code>, with any orphaned sequences all together
in <code>orphans.qc.fq.gz</code>.</p>
<p>Make these trimmed reads read-only and keep them, as we will reuse them later.</p>
<pre><code>chmod a-w ${PROJECT}/trim/*.qc.fq.gz
</code></pre>

<p>Questions:</p>
<ul>
<li>How do you figure out what the parameters mean?</li>
<li>How do you figure out what parameters to use?</li>
<li>What adapters do you use?</li>
<li>What version of Trimmomatic are we using here? (And FastQC?)</li>
<li>Do you think parameters are different for RNAseq and genomic data sets?</li>
<li>What's with these annoyingly long and complicated filenames?</li>
<li>why are we running R1 and R2 together?</li>
</ul>
<p>For a discussion of optimal trimming strategies, see
<a href="http://journal.frontiersin.org/Journal/10.3389/fgene.2014.00013/abstract">MacManes, 2014</a>
-- it's about RNAseq but similar arguments should apply to metagenome
assembly.</p>
<p>Links:</p>
<ul>
<li><a href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic</a></li>
</ul>
<p>Questions:</p>
<ul>
<li>is the quality trimmed data "better" than before?</li>
<li>Does it matter that you still have adapters!?</li>
</ul>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../tara-sample-data/" title="TARA Sample Data" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                TARA Sample Data
              </span>
            </div>
          </a>
        
        
          <a href="../metatranscriptome-assembly/" title="Assembly" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Assembly
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright &copy; 2018 <a href="http://ivory.idyll.org/lab/">Lab for Data Intensive Biology</a> at UC Davis
          </div>
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/application.e72fd936.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:".."}})</script>
      
        <script src="../search/main.js"></script>
      
    
    
      
    
  </body>
</html>