



<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
      
        <link rel="canonical" href="https://ngs-docs.github.io/2018-cicese-metatranscriptomics/sourmash-taxonomic-classification/">
      
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="..">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-3.0.3">
    
    
      
        <title>Sourmash Taxonomic Classification - 2018 CICESE Metatranscriptomics workshop</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/application.451f80e5.css">
      
        <link rel="stylesheet" href="../assets/stylesheets/application-palette.22915126.css">
      
      
        
        
        <meta name="theme-color" content="#009688">
      
    
    
      <script src="../assets/javascripts/modernizr.1aa3b519.js"></script>
    
    
      <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
      
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../assets/fonts/material-icons.css">
    
      <link rel="stylesheet" href="../css/custom.css">
    
    
  </head>
  
    
    
    <body dir="ltr" data-md-color-primary="teal" data-md-color-accent="teal">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448"
    viewBox="0 0 416 448" id="__github">
  <path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19-18.125
        8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19 18.125-8.5
        18.125 8.5 10.75 19 3.125 20.5zM320 304q0 10-3.125 20.5t-10.75
        19-18.125 8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19
        18.125-8.5 18.125 8.5 10.75 19 3.125 20.5zM360
        304q0-30-17.25-51t-46.75-21q-10.25 0-48.75 5.25-17.75 2.75-39.25
        2.75t-39.25-2.75q-38-5.25-48.75-5.25-29.5 0-46.75 21t-17.25 51q0 22 8
        38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0
        37.25-1.75t35-7.375 30.5-15 20.25-25.75 8-38.375zM416 260q0 51.75-15.25
        82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5-41.75
        1.125q-19.5 0-35.5-0.75t-36.875-3.125-38.125-7.5-34.25-12.875-30.25-20.25-21.5-28.75q-15.5-30.75-15.5-82.75
        0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25
        30.875q36.75-8.75 77.25-8.75 37 0 70 8 26.25-20.5
        46.75-30.25t47.25-9.75q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34
        99.5z" />
</svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="../#taxonomic-classification-with-sourmash" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://ngs-docs.github.io/2018-cicese-metatranscriptomics" title="2018 CICESE Metatranscriptomics workshop" class="md-header-nav__button md-logo">
          
            <i class="md-icon">dns</i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            
              <span class="md-header-nav__topic">
                2018 CICESE Metatranscriptomics workshop
              </span>
              <span class="md-header-nav__topic">
                Sourmash Taxonomic Classification
              </span>
            
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  


  <a href="https://github.com/ngs-docs/2018-cicese-metatranscriptomics" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#__github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      2018-cicese-metatranscriptomics
    </div>
  </a>

          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="https://ngs-docs.github.io/2018-cicese-metatranscriptomics" title="2018 CICESE Metatranscriptomics workshop" class="md-nav__button md-logo">
      
        <i class="md-icon">dns</i>
      
    </a>
    2018 CICESE Metatranscriptomics workshop
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/ngs-docs/2018-cicese-metatranscriptomics" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#__github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      2018-cicese-metatranscriptomics
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-1" type="checkbox" id="nav-1">
    
    <label class="md-nav__link" for="nav-1">
      Overview
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-1">
        Overview
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href=".." title="Schedule" class="md-nav__link">
      Schedule
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../code-of-conduct/" title="Code of Conduct" class="md-nav__link">
      Code of Conduct
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2" checked>
    
    <label class="md-nav__link" for="nav-2">
      Day 1
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Day 1
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../welcome/" title="Bienvenidos" class="md-nav__link">
      Bienvenidos
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../cicese-cluster/" title="Cluster login" class="md-nav__link">
      Cluster login
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../setting-up-tara-environment/" title="Software setup" class="md-nav__link">
      Software setup
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../tara-sample-data/" title="TARA Sample Data" class="md-nav__link">
      TARA Sample Data
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../short-read-quality-control/" title="Short Read Quality Control" class="md-nav__link">
      Short Read Quality Control
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        Sourmash Taxonomic Classification
      </label>
    
    <a href="./" title="Sourmash Taxonomic Classification" class="md-nav__link md-nav__link--active">
      Sourmash Taxonomic Classification
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#other-notes" title="Other notes" class="md-nav__link">
    Other notes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#this-is-a-good-way-to-detect-contamination-or-wrong-data" title="This is a good way to detect contamination or wrong data!" class="md-nav__link">
    This is a good way to detect contamination or wrong data!
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../sample-comparison/" title="Comparing Samples" class="md-nav__link">
      Comparing Samples
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../khmer-trimming/" title="Khmer Error Trimming" class="md-nav__link">
      Khmer Error Trimming
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../megahit-assembly/" title="Assembly with Megahit" class="md-nav__link">
      Assembly with Megahit
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../count_transcriptomes/" title="Count Transcriptomes" class="md-nav__link">
      Count Transcriptomes
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3">
    
    <label class="md-nav__link" for="nav-3">
      Day 2
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        Day 2
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../evaluation-and-mapping/" title="Assembly Evaluation" class="md-nav__link">
      Assembly Evaluation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../annotation/" title="Assembly Annotation" class="md-nav__link">
      Assembly Annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../working-with-bioconda/" title="Bioconda" class="md-nav__link">
      Bioconda
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#other-notes" title="Other notes" class="md-nav__link">
    Other notes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#this-is-a-good-way-to-detect-contamination-or-wrong-data" title="This is a good way to detect contamination or wrong data!" class="md-nav__link">
    This is a good way to detect contamination or wrong data!
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                <h1 id="taxonomic-classification-with-sourmash">Taxonomic Classification with Sourmash</h1>
<p>Knowing what species are in our metatranscriptome allows us to take
advantage of other resources already developed for those species, such
as genome sequences and genome annotations.</p>
<p>We will use a tool called <a href="https://sourmash.readthedocs.io/en/latest/">sourmash</a> to get a first pass look at what's in our metatranscriptome.</p>
<p>Sourmash works by calculating a "signature" for a sequence. For taxonomic 
classification, you can then compare that signature to signatures made
from publicly available data. To make this comparison faster, we put the
publicly available sequences in a database that we can compare against. </p>
<p>We have made databases from eukaryotic RNA sequences available in GenBank, 
as well as the MMETSP transcriptomes that 
<a href="https://figshare.com/articles/Marine_Microbial_Eukaryotic_Transcriptome_Sequencing_Project_re-assemblies/3840153">were recently reassembled</a>. 
After making signatures for our assembly, we will search against all of 
our databases at once. </p>
<p>Note that sourmash cannot find matches across large evolutionary distances.
sourmash seems to work well to search and compare data sets for matches 
at the species and genus level, but does not have much sensitivity beyond 
that. It seems to be particularly good at strain-level analysis. You should 
use protein-based analyses to do searches across larger evolutionary distances
(e.g. with a tool like <a href="https://www.kegg.jp/ghostkoala/">KEGG GhostKOALA</a>).</p>
<p>Let's get started!</p>
<p>First, let's make a directory and link in our assembly. We are going to work
with a metatranscriptome assembly that we'll show you how to run later - here we link in a copy that we placed on the cluster already. </p>
<pre><code>mkdir -p ${PROJECT}/sourmash-gather
cd ${PROJECT}/sourmash-gather
ln -s /LUSTRE/bioinformatica_data/bioinformatica2018/assembly/tara_f135_full_megahit.fasta .
</code></pre>

<p>Next, let's make a signature of our assembly.</p>
<p>sourmash uses k-mers to calculate signatures. A k-mer size of 21 is approximately
specific at the genus level, a 31 is at the species level, and 51 at the strain 
level. We will calculate our signature with all three k-mer sizes so we can 
choose which one we want to use later. </p>
<p>The <code>--scaled</code> parameter in the command tells sourmash to only look at 1/10,000th of 
the unique k-mer space. This parameter is agnostic to sequence content, but is 
consistent across signatures. </p>
<p>Lastly, we use the <code>--track-abundance</code> flag. This tells sourmash to keep track of
how often it sees a k-mer. This is meaningful for transcriptomic data because 
expression levels are variable.  </p>
<pre><code>sourmash compute -k 21,31,51 --scaled 10000 --track-abundance -o tara_f135_full_megahit.sig tara_f135_full_megahit.fasta
</code></pre>

<p>Now let's download some databases that contain signatures from transcriptomes of 
publicly-available eukaryotic sequences. We will download databases for plants, 
fungi, protozoa, invertebrates, and vertebrates, as well a database that contains 
sequences from the MMETSP re-assembly project (&gt;700 marine eukaryotic transcriptomes). 
Even if you don't think a certain class of eukaryote is likely to be in your sample
(for instance, vertebrate_mammalian), it's a good idea to include all of the databases
anyway. It can help you quickly locate potential contamination in your sequencing (we
wouldn't expect to find mouse RNA in the ocean, but it is a commonly sequenced organism
and thus if you find it, if may be a contaminant). </p>
<pre><code>ln -fs /LUSTRE/bioinformatica_data/bioinformatica2018/sourmash_databases/genbank-rna-vertebrate_other-k31.tar.gz .
ln -fs /LUSTRE/bioinformatica_data/bioinformatica2018/sourmash_databases/genbank-rna-vertebrate_mammalian-k31.tar.gz .
ln -fs /LUSTRE/bioinformatica_data/bioinformatica2018/sourmash_databases/genbank-rna-invertebrate-k31.tar.gz .
ln -fs /LUSTRE/bioinformatica_data/bioinformatica2018/sourmash_databases/genbank-rna-fungi-k31.tar.gz .
ln -fs /LUSTRE/bioinformatica_data/bioinformatica2018/sourmash_databases/genbank-rna-plant-k31.tar.gz .
ln -fs /LUSTRE/bioinformatica_data/bioinformatica2018/sourmash_databases/genbank-rna-protozoa-k31.tar.gz .
ln -fs /LUSTRE/bioinformatica_data/bioinformatica2018/sourmash_databases/mmetsp-k31-named.tar.gz .
</code></pre>

<blockquote>
<p>Note, if you want to download these files instead, do:</p>
</blockquote>
<pre><code>wget -O genbank-rna-vertebrate_other-k31.tar.gz https://osf.io/qgyax/download
wget -O genbank-rna-vertebrate_mammalian-k31.tar.gz https://osf.io/6c9uy/download
wget -O genbank-rna-invertebrate-k31.tar.gz https://osf.io/7v8ck/download
wget -O genbank-rna-fungi-k31.tar.gz https://osf.io/g6mcr/download
wget -O genbank-rna-plant-k31.tar.gz https://osf.io/kctus/download
wget -O genbank-rna-protozoa-k31.tar.gz https://osf.io/fnu2q/download
wget -O mmetsp-k31-named.tar.gz https://osf.io/cdvqn/download
</code></pre>

<blockquote>
<p>but this will not work on the nodes on the cicese cluster!</p>
</blockquote>
<p>Next, we need to uncompress the databases:</p>
<pre><code>for infile in *.tar.gz
do
    tar xf ${infile}
done
</code></pre>

<p>And now we can perform taxonomic classification!</p>
<pre><code>sourmash gather -k 31 --scaled 10000 -o tara_f135_full_megahit.csv \
    tara_f135_full_megahit.sig \
    *.sbt.json */*.sbt.json
</code></pre>

<p>You should see an output that looks something like this:</p>
<pre><code>overlap     p_query p_match avg_abund
---------   ------- ------- ---------
4.3 Mbp        3.3%    5.5%       1.0    Neoceratium fusus PA161109 MMETSP1074
1.2 Mbp        0.9%    3.8%       1.0    Calcidiscus leptoporus RCC1130 MMETSP...
4.2 Mbp        0.2%    0.3%       1.1    Neoceratium fusus PA161109 MMETSP1075
130.0 kbp      0.1%    0.2%       1.0    Brandtodinium nutricula RCC3387 MMETS...
90.0 kbp       0.1%    0.3%       1.0    Emiliania huxleyi CCMP370 MMETSP1157
70.0 kbp       0.0%    0.3%       1.0    Pelagomonas calceolata RCC969 MMETSP1328
70.0 kbp       0.1%    0.5%       1.2    Prasinoderma singularis RCC927 MMETSP...
80.0 kbp       0.0%    0.1%       1.2    Protoceratium reticulatum CCCM535=CCM...
50.0 kbp       0.0%    0.0%       1.2    Symbiodinium sp. CCMP421 MMETSP1110
50.0 kbp       0.0%    0.1%       1.0    Heterocapsa rotundata SCCAPK-0483 MME...
50.0 kbp       0.0%    0.0%       1.5    Scrippsiella trochoidea CCMP3099 MMET...
60.0 kbp       0.0%    0.1%       1.5    Prorocentrum minimum CCMP2233 MMETSP0269
found less than 40.0 kbp in common. =&gt; exiting

found 12 matches total;
the recovered matches hit 4.9% of the query
</code></pre>

<p>This will take about 5 to 10 minutes to run.</p>
<p>The two columns to pay attention to are <code>p_query</code> and <code>p_match</code>.
<code>p_query</code> is the percent of the metatranscriptome assembly that is
(estimated to be) from the named organism.  <code>p_match</code> is the percent
of the matched transcriptome that is found in the query.  These numbers
are decreased by both evolutionary distance AND by low coverage of the
organism's gene set (low sequencing coverage, or little expression).</p>
<p>sourmash also outputs a csv with this and other information. We will use this csv
to visualize our results later.</p>
<p>We just ran sourmash on our assembly to give us an idea of the taxonomic 
make up of our sample in relation to known sequences. We could also run 
sourmash on our raw reads. This would give us an idea of the number of 
organisms we fail to assemble, or the amount of taxonomic diversity that
is missing from our assembly. Our sample has over 34 million reads and running
gather on a sample this size would take about an hour or two. We ran sourmash like
we did above, but with the reads, before this workshop. We will now download the
csv and compare the results from the assembly and from the reads. </p>
<p>First, link in the gather results from the raw reads. In the future, you can download them
from <a href="https://raw.githubusercontent.com/ngs-docs/2018-cicese-metatranscriptomics/master/docs/files/ERR1719497_paired_gather_all.csv">here</a></p>
<pre><code>ln -s /LUSTRE/bioinformatica_data/bioinformatica2018/scripts/ERR1719497_paired_gather_all.csv .  
</code></pre>

<p>Let's also download a script that we will use to plot the results. 
There are many visualizations we could use, however here we will 
use an upset plot. Upset plots are similar to Venn diagrams, but will
work with many samples. In the future, you can download this script from 
<a href="https://raw.githubusercontent.com/ngs-docs/2018-cicese-metatranscriptomics/master/scripts/plot-gather.py">here</a></p>
<pre><code>ln -s /LUSTRE/bioinformatica_data/bioinformatica2018/scripts/plot-gather.py .

python plot-gather.py
</code></pre>

<p>this will produce a file 'plot-gather.png' that we need to download in order
to visualize.
<a href="https://github.com/ngs-docs/2018-cicese-metatranscriptomics/blob/master/docs/files/plot-gather.png">Here</a> is a copy of this output
visualization.</p>
<p>We can see that the reads have more matches than the assembly. </p>
<h2 id="other-notes">Other notes</h2>
<p>We used the raw reads to do taxonomic classification. Although the errors
present in the raw reads may slow sourmash down a little, there is a chance 
when error trimming transcriptomes that real biological variation is removed. 
Because sourmash is so specific, the errors don't give false positives and so 
keeping all of the possible real variation could improve taxonomic recall in 
some cases.</p>
<h2 id="this-is-a-good-way-to-detect-contamination-or-wrong-data">This is a good way to detect contamination or wrong data!</h2>
<p>We recommend doing <code>sourmash gather</code> immediately after receiving your
data from the sequencing facility. If your environmental metagenome has a
tremendous amount of mouse sequence in it... maybe the sequencing facility
sent you the wrong data?</p>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../short-read-quality-control/" title="Short Read Quality Control" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Short Read Quality Control
              </span>
            </div>
          </a>
        
        
          <a href="../sample-comparison/" title="Comparing Samples" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Comparing Samples
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright &copy; 2018 <a href="http://ivory.idyll.org/lab/">Lab for Data Intensive Biology</a> at UC Davis
          </div>
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/application.e72fd936.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:".."}})</script>
      
        <script src="../search/main.js"></script>
      
    
    
      
    
  </body>
</html>