



<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
      
        <link rel="canonical" href="https://ngs-docs.github.io/2018-cicese-metatranscriptomics/evaluation-and-mapping/">
      
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching pages">
      
        <meta name="lang:search.result.one" content="1 matching page">
      
        <meta name="lang:search.result.other" content="# matching page">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="..">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-3.0.3">
    
    
      
        <title>Assembly Evaluation and Quantification - 2018 CICESE Metatranscriptomics workshop</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/application.451f80e5.css">
      
        <link rel="stylesheet" href="../assets/stylesheets/application-palette.22915126.css">
      
      
        
        
        <meta name="theme-color" content="#009688">
      
    
    
      <script src="../assets/javascripts/modernizr.1aa3b519.js"></script>
    
    
      <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
      
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../assets/fonts/material-icons.css">
    
      <link rel="stylesheet" href="../css/custom.css">
    
    
  </head>
  
    
    
    <body dir="ltr" data-md-color-primary="teal" data-md-color-accent="teal">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448"
    viewBox="0 0 416 448" id="__github">
  <path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19-18.125
        8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19 18.125-8.5
        18.125 8.5 10.75 19 3.125 20.5zM320 304q0 10-3.125 20.5t-10.75
        19-18.125 8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19
        18.125-8.5 18.125 8.5 10.75 19 3.125 20.5zM360
        304q0-30-17.25-51t-46.75-21q-10.25 0-48.75 5.25-17.75 2.75-39.25
        2.75t-39.25-2.75q-38-5.25-48.75-5.25-29.5 0-46.75 21t-17.25 51q0 22 8
        38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0
        37.25-1.75t35-7.375 30.5-15 20.25-25.75 8-38.375zM416 260q0 51.75-15.25
        82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5-41.75
        1.125q-19.5 0-35.5-0.75t-36.875-3.125-38.125-7.5-34.25-12.875-30.25-20.25-21.5-28.75q-15.5-30.75-15.5-82.75
        0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25
        30.875q36.75-8.75 77.25-8.75 37 0 70 8 26.25-20.5
        46.75-30.25t47.25-9.75q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34
        99.5z" />
</svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="../#set-up-the-directory" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://ngs-docs.github.io/2018-cicese-metatranscriptomics" title="2018 CICESE Metatranscriptomics workshop" class="md-header-nav__button md-logo">
          
            <i class="md-icon">dns</i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            
              <span class="md-header-nav__topic">
                2018 CICESE Metatranscriptomics workshop
              </span>
              <span class="md-header-nav__topic">
                Assembly Evaluation and Quantification
              </span>
            
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search this site" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching this site
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  


  <a href="https://github.com/ngs-docs/2018-cicese-metatranscriptomics" title="Go to repository" class="md-source" data-md-source="github">
    
    <div class="md-source__repository">
      2018-cicese-metatranscriptomics
    </div>
  </a>

          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="https://ngs-docs.github.io/2018-cicese-metatranscriptomics" title="2018 CICESE Metatranscriptomics workshop" class="md-nav__button md-logo">
      
        <i class="md-icon">dns</i>
      
    </a>
    2018 CICESE Metatranscriptomics workshop
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/ngs-docs/2018-cicese-metatranscriptomics" title="Go to repository" class="md-source" data-md-source="github">
    
    <div class="md-source__repository">
      2018-cicese-metatranscriptomics
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-1" type="checkbox" id="nav-1">
    
    <label class="md-nav__link" for="nav-1">
      Overview
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-1">
        Overview
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href=".." title="Schedule" class="md-nav__link">
      Schedule
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../code-of-conduct/" title="Code of Conduct" class="md-nav__link">
      Code of Conduct
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2">
    
    <label class="md-nav__link" for="nav-2">
      Day 1
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Day 1
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../welcome/" title="Bienvenidos" class="md-nav__link">
      Bienvenidos
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../cicese-cluster/" title="Cluster login" class="md-nav__link">
      Cluster login
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../setting-up-tara-environment/" title="Software setup" class="md-nav__link">
      Software setup
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../tara-sample-data/" title="TARA Sample Data" class="md-nav__link">
      TARA Sample Data
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../short-read-quality-control/" title="Short Read Quality Control" class="md-nav__link">
      Short Read Quality Control
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../sourmash-taxonomic-classification/" title="Sourmash Taxonomic Classification" class="md-nav__link">
      Sourmash Taxonomic Classification
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../sample-comparison/" title="Comparing Samples" class="md-nav__link">
      Comparing Samples
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3" checked>
    
    <label class="md-nav__link" for="nav-3">
      Day 2
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        Day 2
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../khmer-trimming/" title="Khmer Error Trimming" class="md-nav__link">
      Khmer Error Trimming
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../megahit-assembly/" title="Assembly with Megahit" class="md-nav__link">
      Assembly with Megahit
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../count_transcriptomes/" title="Count Transcriptomes" class="md-nav__link">
      Count Transcriptomes
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
    <a href="./" title="Assembly Evaluation and Quantification" class="md-nav__link md-nav__link--active">
      Assembly Evaluation and Quantification
    </a>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../annotation/" title="Assembly Annotation" class="md-nav__link">
      Assembly Annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../working-with-bioconda/" title="Bioconda" class="md-nav__link">
      Bioconda
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                <p>How well did our assembly work?</p>
<h2 id="set-up-the-directory">Set up the directory</h2>
<p>First, make sure you have the <code>PROJECT</code> variable defined:</p>
<pre><code>echo $PROJECT
</code></pre>

<p>if you don't see any output, make sure to redefine the $PROJECT variable.</p>
<p>Now, let's create a folder <code>evaluation</code> to work in:</p>
<pre><code>cd $PROJECT
mkdir -p evaluation
cd evaluation
</code></pre>

<p>Now, let's copy in the assembly we made in the previous lesson:</p>
<pre><code>cp $PROJECT/assembly/tara135_SRF_megahit.fasta ./
</code></pre>

<h2 id="generate-assembly-statistics-with-transrate">Generate assembly statistics with Transrate</h2>
<p><a href="http://hibberdlab.com/transrate/getting_started.html">Transrate</a> is a program that can be used for a couple different types of assembly evaluation. The first and simplest method is to calculate length-based metrics about the assembly, such as the total number of bases, and the N50 of the contigs. Transrate can also be used to compare two assemblies or give you a score which represents proportion of input reads that provide positive support for the assembly. For a further explanation of metrics and how to run the reference-based transrate, see the <a href="http://hibberdlab.com/transrate/metrics.html">documentation</a> and the paper by <a href="http://genome.cshlp.org/content/early/2016/06/01/gr.196469.115">Smith-Unna et al. 2016</a>. </p>
<p>We have installed transrate for you. However, if you need to install it in the future, see installation instructions <a href="../setting-up-tara-environment/">here</a>.</p>
<p>See options for running transrate:</p>
<pre><code>transrate -h
</code></pre>

<p>Let's use transrate to calculate some stats on our assembly contigs:</p>
<pre><code>transrate --assembly tara135_SRF_megahit.fasta
</code></pre>

<p>You should see output that looks like this:</p>
<pre><code>[ INFO] 2018-11-06 23:50:35 : Loading assembly: /LUSTRE/bioinformatica_data/bioinformatica2018/assembly/tara135_SRF_megahit.fasta
[ INFO] 2018-11-06 23:50:35 : Analysing assembly: /LUSTRE/bioinformatica_data/bioinformatica2018/assembly/tara135_SRF_megahit.fasta
[ INFO] 2018-11-06 23:50:35 : Results will be saved in /LUSTRE/bioinformatica_data/bioinformatica2018/assembly/transrate_results/tara135_SRF_megahit
[ INFO] 2018-11-06 23:50:35 : Calculating contig metrics...
[ INFO] 2018-11-06 23:50:35 : Contig metrics:
[ INFO] 2018-11-06 23:50:35 : -----------------------------------
[ INFO] 2018-11-06 23:50:35 : n seqs                         1502
[ INFO] 2018-11-06 23:50:35 : smallest                        200
[ INFO] 2018-11-06 23:50:35 : largest                        4998
[ INFO] 2018-11-06 23:50:35 : n bases                      638347
[ INFO] 2018-11-06 23:50:35 : mean len                      425.0
[ INFO] 2018-11-06 23:50:35 : n under 200                       0
[ INFO] 2018-11-06 23:50:35 : n over 1k                        40
[ INFO] 2018-11-06 23:50:35 : n over 10k                        0
[ INFO] 2018-11-06 23:50:35 : n with orf                      331
[ INFO] 2018-11-06 23:50:35 : mean orf percent              83.54
[ INFO] 2018-11-06 23:50:35 : n90                             232
[ INFO] 2018-11-06 23:50:35 : n70                             360
[ INFO] 2018-11-06 23:50:35 : n50                             453
[ INFO] 2018-11-06 23:50:35 : n30                             599
[ INFO] 2018-11-06 23:50:35 : n10                             935
[ INFO] 2018-11-06 23:50:35 : gc                             0.51
[ INFO] 2018-11-06 23:50:35 : bases n                           0
[ INFO] 2018-11-06 23:50:35 : proportion n                    0.0
[ INFO] 2018-11-06 23:50:35 : Contig metrics done in 0 seconds
[ INFO] 2018-11-06 23:50:35 : No reads provided, skipping read diagnostics
[ INFO] 2018-11-06 23:50:35 : No reference provided, skipping comparative diagnostics
[ INFO] 2018-11-06 23:50:35 : Writing contig metrics for each contig to /LUSTRE/bioinformatica_data/bioinformatica2018/assembly/transrate_results/tara135_SRF_megahit/contigs.csv
[ INFO] 2018-11-06 23:50:35 : Writing analysis results to assemblies.csv
</code></pre>

<h2 id="comparing-assemblies">Comparing Assemblies</h2>
<p>We built a metatranscriptome with the full set of TARA_SRF reads. Copy this into your evaluation directory</p>
<pre><code>cd ${PROJECT}/evaluation
ln -s LUSTRE/bioinformatica_data/bioinformatica2018/assembly/tara_f135_full_megahit.fasta
</code></pre>

<ul>
<li>How do the two transcriptomes compare with each other?</li>
</ul>
<pre><code># full vs. subset
transrate --reference=tara_f135_full_megahit.fasta --assembly=tara135_SRF_megahit.fasta --output=full_v_subset

# subset vs. full
transrate --assembly=tara_f135_full_megahit.fasta --reference=tara135_SRF_megahit.fasta --output=subset_v_full
</code></pre>

<h2 id="how-well-does-this-assembly-represent-our-sequenced-reads">How well does this assembly represent our sequenced reads?</h2>
<p>It's useful to know how well the transcripts represent the sequenced reads. To do this, we'll need to link in the reads we used to generate this assembly: </p>
<pre><code>ln -s ${PROJECT}/trimmed/TARA_135_SRF_5-20_*.qc.fq.gz ./
</code></pre>

<h1 id="quantifying-reads-with-salmon">Quantifying reads with Salmon</h1>
<p>We will use <a href="http://salmon.readthedocs.org/en/latest/">Salmon</a> to
quantify expression. Salmon is a new breed of software for quantifying RNAseq reads that is both really fast and takes
transcript length into consideration (<a href="https://www.nature.com/articles/nmeth.4197">Patro et al. 2017</a>). </p>
<p>You can read more about salmon-like "pseudoalignment" here: </p>
<ul>
<li>Intro blog post: http://robpatro.com/blog/?p=248</li>
<li>A 2016 blog post evaluating and comparing methods <a href="https://cgatoxford.wordpress.com/2016/08/17/why-you-should-stop-using-featurecounts-htseq-or-cufflinks2-and-start-using-kallisto-salmon-or-sailfish/">here</a></li>
<li>Salmon github repo <a href="https://github.com/COMBINE-lab/salmon">here</a></li>
<li>2018 paper: <a href="https://www.biorxiv.org/content/early/2018/10/16/444620">A direct comparison of genome alignment and transcriptome pseudoalignment</a></li>
</ul>
<h2 id="quantification-with-salmon">Quantification with Salmon</h2>
<p>Check that salmon is available and see run options:</p>
<pre><code>salmon -h
</code></pre>

<p>Now let's check that we still have the trimmed data we created day 1:</p>
<pre><code>set -u
printf &quot;\nMy trimmed data is in $PROJECT/quality/, and consists of $(ls -1 ${PROJECT}/quality/*.qc.fq.gz | wc -l) files\n\n&quot;
set +u

</code></pre>

<p>where set -u should let you know if you have any unset variables, i.e. if the <code>$PROJECT</code> variable is not defined. </p>
<p>If you see <code>-bash: PROJECT: unbound variable</code>, then you need to set the $PROJECT variable.  </p>
<pre><code>export PROJECT= ~/work
</code></pre>

<p>and then re-run the <code>printf</code> code block.</p>
<p>NOTE: if you do not have files, please rerun quality trimming steps <a href="../short-read-quality-control/">here</a></p>
<h2 id="create-a-directory-to-work-in">Create a directory to work in</h2>
<pre><code>   cd ${PROJECT}
   mkdir -p quant
   cd quant
</code></pre>

<h2 id="link-an-assembly">Link an assembly</h2>
<p>We use a full assembly to use for mapping. This assembly was made with all TARA_135 SRF reads, rather than the subset we used in the <a href="../megahit-assembly/">assembly</a> tutorial.</p>
<pre><code>ln -s /LUSTRE/bioinformatica_data/bioinformatica2018/assembly/tara_f135_full_megahit.fasta ./
</code></pre>

<h2 id="run-salmon">Run Salmon</h2>
<p>Build an index for the metatranscriptome:</p>
<pre><code>    salmon index --index tara135 --transcripts tara_f135_full_megahit.fasta
</code></pre>

<p>Next, link in the QC reads (produced in <a href="../short-read-quality-control/">quality</a>:</p>
<pre><code>   ln -s ../quality/*.qc.fq.gz ./
</code></pre>

<p>Then, run the salmon mapping:</p>
<pre><code>for sample in *1.qc.fq.gz
do
  base=$(basename $sample _1.fq.gz)
  echo $base
  salmon quant -i tara135 -p 2 -l A -1 ${base}_1.qc.fq.gz -2 ${base}_2.qc.fq.gz -o ${base}_quant
done
</code></pre>

<p>This will create a bunch of directories named something like
<code>TARA_135_SRF_5-20_rep1_quant</code>, containing a bunch of files. Take a
look at what files there are:</p>
<p>You should see:</p>
<pre><code>   aux_info
   cmd_info.json
   lib_format_counts.json
   libParams
   logs
   quant.sf
</code></pre>

<p>The two most interesting files are <code>quant.sf</code>, which contains the counts,
and <code>salmon_quant.log</code> (in the <code>logs</code> directory), which contains a
log from the salmon run.</p>
<h1 id="looking-at-count-data">Looking at count data</h1>
<p>The <code>quant.sf</code> files contain the relevant information about
contig expression -- take a look</p>
<pre><code>   head -20 TARA_135_SRF_5-20_rep1_1m_quant/quant.sf 
</code></pre>

<p>You should see output that looks like this:</p>
<pre><code>Name    Length  EffectiveLength TPM NumReads
k119_5  212 63.757  0.000000    0.000
k119_10 231 75.730  0.000000    0.000
k119_14 261 97.683  0.000000    0.000
k119_16 301 130.690 11.736728   1.000
k119_18 302 131.560 0.000000    0.000
k119_21 203 58.357  0.000000    0.000
k119_22 308 136.790 0.000000    0.000
</code></pre>

<p>The first column contains the transcript names, and the
fifth column is the "raw counts", which is what many 
differential expression programs need.</p>
<h1 id="assess-mapping-rate">Assess Mapping rate</h1>
<p>Okay, we got some numbers for how much the transcripts were expressed, but how well did the reads map overall? 
Let's go look at one of the log files to see.</p>
<pre><code>less TARA_135_SRF_5-20_rep1_1m_quant/logs/salmon_quant.log

</code></pre>

<p>How well did the reads map to the metatranscriptome?</p>
<p>Let's see how the rest of the files mapped. We can 
look at the mapping rate in each log file by executing
the following:</p>
<pre><code>grep Mapping *quant/logs/*
</code></pre>

<p>grep is a handy program that finds and prints lines in files that match a pattern. In this case, 
the pattern we're searching for is the word 'Mapping', and we're searching in any directory that 
ends in <code>quant</code> and has a logs directory with at least one file in it.</p>
<p>How do the mapping rates compare? What does this tell us about our metatranscriptome?</p>
<h2 id="how-well-does-quantification-capture-sample-distances">How well does quantification capture sample distances?</h2>
<p>We made an MDS plot of our <code>sourmash compare</code> results yesterday. With our salmon
<code>quant.sf</code> files, we can also make and MDS plot and see how different the two are.
This will demonstrate how well mapping to our assembly captures the information
in our reads. </p>
<p>We already ran the code to do this, but if you want to see what it looks like, 
you can find it <a href="https://raw.githubusercontent.com/ngs-docs/2018-cicese-metatranscriptomics/master/scripts/salmon_mds_plot.R">here</a>. It relies on the <code>quant.sf</code> files output by Salmon. </p>
<p><img alt="sourmash" src="../files/tara-trimmed-comp-mds.png" />
<img alt="salmon" src="../files/tara-salmon-mds.png" /></p>
<p>Look how different they are! To be fair, we only mapped back to a full transcriptome
with 1 million reads, but this is still a good test. </p>
<p>What might this say about our samples?</p>
<h2 id="another-way-to-assess-read-mapping">Another way to assess read mapping</h2>
<p>Transrate actually  has a <code>read assessment</code> mode that uses <code>salmon</code> to "align" reads to the transcriptome and generates some metrics on read mapping. </p>
<p>We won't run this today, but you could run it via:</p>
<pre><code>transrate --assembly=tara135_SRF_megahit.fasta --threads=2 --left=*_1.qc.fq.gz --right *_2.qc.fq.gz --output=${PROJECT}/evaluation/tara135_SRF_transrate
</code></pre>

<h2 id="other-useful-tutorials-and-references">Other useful tutorials and references</h2>
<ul>
<li>https://github.com/ngs-docs/2015-nov-adv-rna/blob/master/salmon.rst</li>
<li>http://angus.readthedocs.io/en/2016/rob_quant/tut.html</li>
<li>https://2016-aug-nonmodel-rnaseq.readthedocs.io/en/latest/quantification.html</li>
</ul>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../count_transcriptomes/" title="Count Transcriptomes" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Count Transcriptomes
              </span>
            </div>
          </a>
        
        
          <a href="../annotation/" title="Assembly Annotation" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Assembly Annotation
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright &copy; 2018 <a href="http://ivory.idyll.org/lab/">Lab for Data Intensive Biology</a> at UC Davis
          </div>
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/application.e72fd936.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:".."}})</script>
      
        <script src="../search/main.js"></script>
      
    
    
      
    
  </body>
</html>