



<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
      
        <link rel="canonical" href="https://ngs-docs.github.io/2018-cicese-metatranscriptomics/khmer-trimming-viz-2017/">
      
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching pages">
      
        <meta name="lang:search.result.one" content="1 matching page">
      
        <meta name="lang:search.result.other" content="# matching page">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="..">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-3.0.3">
    
    
      
        <title>K-mer Error Trimming - 2018 CICESE Metatranscriptomics workshop</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/application.451f80e5.css">
      
        <link rel="stylesheet" href="../assets/stylesheets/application-palette.22915126.css">
      
      
        
        
        <meta name="theme-color" content="#009688">
      
    
    
      <script src="../assets/javascripts/modernizr.1aa3b519.js"></script>
    
    
      <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
      
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../assets/fonts/material-icons.css">
    
      <link rel="stylesheet" href="../css/custom.css">
    
    
  </head>
  
    
    
    <body dir="ltr" data-md-color-primary="teal" data-md-color-accent="teal">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448"
    viewBox="0 0 416 448" id="__github">
  <path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19-18.125
        8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19 18.125-8.5
        18.125 8.5 10.75 19 3.125 20.5zM320 304q0 10-3.125 20.5t-10.75
        19-18.125 8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19
        18.125-8.5 18.125 8.5 10.75 19 3.125 20.5zM360
        304q0-30-17.25-51t-46.75-21q-10.25 0-48.75 5.25-17.75 2.75-39.25
        2.75t-39.25-2.75q-38-5.25-48.75-5.25-29.5 0-46.75 21t-17.25 51q0 22 8
        38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0
        37.25-1.75t35-7.375 30.5-15 20.25-25.75 8-38.375zM416 260q0 51.75-15.25
        82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5-41.75
        1.125q-19.5 0-35.5-0.75t-36.875-3.125-38.125-7.5-34.25-12.875-30.25-20.25-21.5-28.75q-15.5-30.75-15.5-82.75
        0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25
        30.875q36.75-8.75 77.25-8.75 37 0 70 8 26.25-20.5
        46.75-30.25t47.25-9.75q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34
        99.5z" />
</svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="../#k-mer-error-trimming" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://ngs-docs.github.io/2018-cicese-metatranscriptomics" title="2018 CICESE Metatranscriptomics workshop" class="md-header-nav__button md-logo">
          
            <i class="md-icon">dns</i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            
              <span class="md-header-nav__topic">
                2018 CICESE Metatranscriptomics workshop
              </span>
              <span class="md-header-nav__topic">
                K-mer Error Trimming
              </span>
            
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search this site" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching this site
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  


  <a href="https://github.com/ngs-docs/2018-cicese-metatranscriptomics" title="Go to repository" class="md-source" data-md-source="github">
    
    <div class="md-source__repository">
      2018-cicese-metatranscriptomics
    </div>
  </a>

          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="https://ngs-docs.github.io/2018-cicese-metatranscriptomics" title="2018 CICESE Metatranscriptomics workshop" class="md-nav__button md-logo">
      
        <i class="md-icon">dns</i>
      
    </a>
    2018 CICESE Metatranscriptomics workshop
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/ngs-docs/2018-cicese-metatranscriptomics" title="Go to repository" class="md-source" data-md-source="github">
    
    <div class="md-source__repository">
      2018-cicese-metatranscriptomics
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-1" type="checkbox" id="nav-1">
    
    <label class="md-nav__link" for="nav-1">
      Overview
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-1">
        Overview
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href=".." title="Schedule" class="md-nav__link">
      Schedule
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../code-of-conduct/" title="Code of Conduct" class="md-nav__link">
      Code of Conduct
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2">
    
    <label class="md-nav__link" for="nav-2">
      Day 1
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Day 1
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../welcome/" title="Bienvenidos" class="md-nav__link">
      Bienvenidos
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../cicese-cluster/" title="Cluster login" class="md-nav__link">
      Cluster login
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../setting-up-tara-environment/" title="Software setup" class="md-nav__link">
      Software setup
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../tara-sample-data/" title="TARA Sample Data" class="md-nav__link">
      TARA Sample Data
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../short-read-quality-control/" title="Short Read Quality Control" class="md-nav__link">
      Short Read Quality Control
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../sourmash-taxonomic-classification/" title="Sourmash Taxonomic Classification" class="md-nav__link">
      Sourmash Taxonomic Classification
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../sample-comparison/" title="Comparing Samples" class="md-nav__link">
      Comparing Samples
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3">
    
    <label class="md-nav__link" for="nav-3">
      Day 2
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        Day 2
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../khmer-trimming/" title="Khmer Error Trimming" class="md-nav__link">
      Khmer Error Trimming
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../megahit-assembly/" title="Assembly with Megahit" class="md-nav__link">
      Assembly with Megahit
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../evaluation-and-mapping/" title="Assembly Evaluation and Quantification" class="md-nav__link">
      Assembly Evaluation and Quantification
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../annotation/" title="Assembly Annotation" class="md-nav__link">
      Assembly Annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../count_transcriptomes/" title="Count Transcriptomes" class="md-nav__link">
      Count Transcriptomes
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../filezilla/" title="Filezilla" class="md-nav__link">
      Filezilla
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../working-with-bioconda/" title="Bioconda" class="md-nav__link">
      Bioconda
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../automation-qc/" title="Workflows and Repeatability" class="md-nav__link">
      Workflows and Repeatability
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../references/" title="References" class="md-nav__link">
      References
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#why-or-why-not-do-k-mer-trimming" title="Why (or why not) do k-mer trimming?" class="md-nav__link">
    Why (or why not) do k-mer trimming?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#set-up-workspace-and-install-khmer" title="Set up workspace and install khmer" class="md-nav__link">
    Set up workspace and install khmer
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assess-changes-in-kmer-abundance" title="Assess changes in kmer abundance" class="md-nav__link">
    Assess changes in kmer abundance
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#visualization-of-khmer-error-trimming-2017-workshop" title="Visualization of Khmer error trimming (2017 workshop)" class="md-nav__link">
    Visualization of Khmer error trimming (2017 workshop)
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#why-or-why-not-do-k-mer-trimming_1" title="Why (or why not) do k-mer trimming?" class="md-nav__link">
    Why (or why not) do k-mer trimming?
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                <h1 id="k-mer-error-trimming">K-mer Error Trimming</h1>
<p>(Optional)</p>
<p>khmer documentation: http://khmer.readthedocs.io/en/latest</p>
<h2 id="why-or-why-not-do-k-mer-trimming">Why (or why not) do k-mer trimming?</h2>
<p>If you can assemble your data set without k-mer trimming, there's no
reason to do it.  The reason we're error trimming here is to speed up
the assembler (by removing data) and to decrease the memory requirements
of the assembler (by removing a number of k-mers).</p>
<h2 id="set-up-workspace-and-install-khmer">Set up workspace and install khmer</h2>
<pre><code>conda install khmer
</code></pre>

<p>To run error trimming, use the khmer script <code>trim-low-abund.py</code>:</p>
<pre><code>cd ${PROJECT}/trim

for filename in *_1.qc.fq.gz
do
  #Use the program basename to remove _1.qc.fq.gz to generate the base
  base=$(basename $filename _1.qc.fq.gz)
  echo $base

  #Run khmer trimming
  (interleave-reads.py ${base}_1.qc.fq.gz ${base}_2.qc.fq.gz )| \
  (trim-low-abund.py - -V -Z 10 -C 3 -o - --gzip -M 8e9) | \ 
  (extract-paired-reads.py --gzip -p ${base}.khmer.pe.fq.gz -s ${base}.khmer.se.fq.gz)

done
</code></pre>

<h2 id="assess-changes-in-kmer-abundance">Assess changes in kmer abundance</h2>
<p>To see how many k-mers we removed, you can examine the distribution as above,
or use the <code>unique-kmers.py</code> script. Let's compare kmers for one sample.</p>
<pre><code>unique-kmers.py TARA_135_SRF_5-20_rep1_1m_1.qc.fq.gz TARA_135_SRF_5-20_rep1_1m_2.qc.fq.gz
unique-kmers.py TARA_135_SRF_5-20_rep1_1m.khmer.pe.fq.gz
</code></pre>

<hr />
<h2 id="visualization-of-khmer-error-trimming-2017-workshop">Visualization of Khmer error trimming (2017 workshop)</h2>
<p>Let's plot a kmer abundance histogram of one sample</p>
<pre><code>cd $PROJECT
abundance-dist-single.py -M 1e9 -k 21 SRR1976948_1.fastq.gz SRR1976948_1.fastq.gz.dist

</code></pre>

<p>If you plot a k-mer abundance histogram of the samples, you'll
notice something: there's an awful lot of unique (abundance=1) k-mers.
These are erroneous k-mers caused by sequencing errors.</p>
<p>and in another cell::</p>
<p>%matplotlib inline
  import numpy
  from pylab import *
  dist1 = numpy.loadtxt('SRR1976948_1.fastq.gz.dist', skiprows=1, delimiter=',')
  plot(dist1[:,0], dist1[:,1])
  axis(xmax=50)</p>
<p>Many of these errors remain even after you do the Trimmomatic run; you can
see this with::</p>
<p>!abundance-dist-single.py -M 1e9 -k 21 SRR1976948_1.qc.fq.gz SRR1976948_1.qc.fq.gz.dist</p>
<p>and then plot::</p>
<p>dist2 = numpy.loadtxt('SRR1976948_1.qc.fq.gz.dist', skiprows=1, delimiter=',')
  plot(dist1[:,0], dist1[:,1], label='untrimmed')
  plot(dist2[:,0], dist2[:,1], label='trimmed')
  legend(loc='upper right')
  axis(xmax=50)</p>
<p>This is for
two reasons:</p>
<p>First, Trimmomatic trims based solely on the quality score, which is
a statistical statement about the correctness of a base - a Q score
of 30 means that, of 1000 bases with that Q score, 1 of those
bases will be wrong.  So, a base can have a high Q score and still
be wrong! (and <strong>many</strong> bases will have a low Q score and still be
correct)</p>
<p>Second, we trimmed <strong>very</strong> lightly - only bases that had a very low
quality were removed.  This was intentional because with assembly,
you want to retain as much coverage as possible, and the assembler
will generally figure out what the "correct" base is from the coverage.</p>
<p>An alternative to trimming based on the quality scores is to trim based on
k-mer abundance - this is known as k-mer spectral error trimming.  K-mer
spectral error trimming <em>always</em> beats quality score trimming in terms
of eliminating errors; e.g. look at this table from <code>Zhang et al., 2014 &lt;http://journals.plos.org/plosone/article?id=10.1371%2Fjournal.pone.0101271&gt;</code>__:</p>
<p>.. thumbnail:: files/2014-zhang.png
   :width: 40%</p>
<p>The basic logic is this: if you see low abundance k-mers in a high
coverage data set, those k-mers are almost certainly the result of
errors.  (Caveat: strain variation could also create them.)</p>
<p>In metagenomic data sets we do have the problem that we may have very
low and very high coverage data.  So we don't necessarily want to get
rid of all low-abundance k-mers, because they may represent truly low
abundance (but useful) data.</p>
<p>As part of the khmer project in my lab, we have developed an approach
that sorts reads into high abundance and low abundance reads, and only
error trims the high abundance reads.</p>
<p>.. thumbnail:: files/kmer-trimming.png
   :width: 40%</p>
<p>This does mean that many errors may get left in the data set, because we
have no way of figuring out if they are errors or simply low coverage,
but that's OK (and you can always trim them off if you really care).</p>
<p>.. Error profile@@</p>
<p>To run such error trimming, use the command <code>trim-low-abund.py</code>
(at the command line, or prefix with a '!' in the notebook)::</p>
<p>interleave-reads.py SRR1976948_1.qc.fq.gz SRR1976948_2.qc.fq.gz |
     trim-low-abund.py -V -M 8e9 -C 3 -Z 10 - -o SRR1976948.trim.fq</p>
<h2 id="why-or-why-not-do-k-mer-trimming_1">Why (or why not) do k-mer trimming?</h2>
<p>If you can assemble your data set without k-mer trimming, there's no
reason to do it.  The reason we're error trimming here is to speed up
the assembler (by removing data) and to decrease the memory requirements
of the assembler (by removing a number of k-mers).</p>
<p>To see how many k-mers we removed, you can examine the distribution as above,
or use the <code>unique-kmers.py</code> script::</p>
<pre><code>unique-kmers.py SRR1976948_1.qc.fq.gz SRR1976948_2.qc.fq.gz
unique-kmers.py SRR1976948.trim.fq
</code></pre>
<hr />
<p>Next: :doc:<code>assemble</code></p>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright &copy; 2018 <a href="http://ivory.idyll.org/lab/">Lab for Data Intensive Biology</a> at UC Davis
          </div>
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/application.e72fd936.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:".."}})</script>
      
        <script src="../search/main.js"></script>
      
    
    
      
    
  </body>
</html>