



<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
      
        <link rel="canonical" href="https://ngs-docs.github.io/2018-cicese-metatranscriptomics/automation/">
      
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="..">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-3.0.3">
    
    
      
        <title>Automation via shell scripts and snakemake - 2018 CICESE Metatranscriptomics workshop</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/application.451f80e5.css">
      
        <link rel="stylesheet" href="../assets/stylesheets/application-palette.22915126.css">
      
      
        
        
        <meta name="theme-color" content="#009688">
      
    
    
      <script src="../assets/javascripts/modernizr.1aa3b519.js"></script>
    
    
      <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
      
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../assets/fonts/material-icons.css">
    
      <link rel="stylesheet" href="../css/custom.css">
    
    
  </head>
  
    
    
    <body dir="ltr" data-md-color-primary="teal" data-md-color-accent="teal">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448"
    viewBox="0 0 416 448" id="__github">
  <path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19-18.125
        8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19 18.125-8.5
        18.125 8.5 10.75 19 3.125 20.5zM320 304q0 10-3.125 20.5t-10.75
        19-18.125 8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19
        18.125-8.5 18.125 8.5 10.75 19 3.125 20.5zM360
        304q0-30-17.25-51t-46.75-21q-10.25 0-48.75 5.25-17.75 2.75-39.25
        2.75t-39.25-2.75q-38-5.25-48.75-5.25-29.5 0-46.75 21t-17.25 51q0 22 8
        38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0
        37.25-1.75t35-7.375 30.5-15 20.25-25.75 8-38.375zM416 260q0 51.75-15.25
        82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5-41.75
        1.125q-19.5 0-35.5-0.75t-36.875-3.125-38.125-7.5-34.25-12.875-30.25-20.25-21.5-28.75q-15.5-30.75-15.5-82.75
        0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25
        30.875q36.75-8.75 77.25-8.75 37 0 70 8 26.25-20.5
        46.75-30.25t47.25-9.75q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34
        99.5z" />
</svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="../#automation-via-shell-scripts-and-snakemake" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://ngs-docs.github.io/2018-cicese-metatranscriptomics" title="2018 CICESE Metatranscriptomics workshop" class="md-header-nav__button md-logo">
          
            <i class="md-icon">dns</i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            
              <span class="md-header-nav__topic">
                2018 CICESE Metatranscriptomics workshop
              </span>
              <span class="md-header-nav__topic">
                Automation via shell scripts and snakemake
              </span>
            
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  


  <a href="https://github.com/ngs-docs/2018-cicese-metatranscriptomics" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#__github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      2018-cicese-metatranscriptomics
    </div>
  </a>

          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="https://ngs-docs.github.io/2018-cicese-metatranscriptomics" title="2018 CICESE Metatranscriptomics workshop" class="md-nav__button md-logo">
      
        <i class="md-icon">dns</i>
      
    </a>
    2018 CICESE Metatranscriptomics workshop
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/ngs-docs/2018-cicese-metatranscriptomics" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#__github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      2018-cicese-metatranscriptomics
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-1" type="checkbox" id="nav-1">
    
    <label class="md-nav__link" for="nav-1">
      Overview
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-1">
        Overview
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href=".." title="Schedule" class="md-nav__link">
      Schedule
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../code-of-conduct/" title="Code of Conduct" class="md-nav__link">
      Code of Conduct
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2">
    
    <label class="md-nav__link" for="nav-2">
      Day 1
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Day 1
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../welcome/" title="Bienvenidos" class="md-nav__link">
      Bienvenidos
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../cicese-cluster/" title="Cluster login" class="md-nav__link">
      Cluster login
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../setting-up-tara-environment/" title="Software setup" class="md-nav__link">
      Software setup
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../tara-sample-data/" title="TARA Sample Data" class="md-nav__link">
      TARA Sample Data
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../short-read-quality-control/" title="Short Read Quality Control" class="md-nav__link">
      Short Read Quality Control
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../sourmash-taxonomic-classification/" title="Sourmash Taxonomic Classification" class="md-nav__link">
      Sourmash Taxonomic Classification
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../sample-comparison/" title="Comparing Samples" class="md-nav__link">
      Comparing Samples
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3">
    
    <label class="md-nav__link" for="nav-3">
      Day 2
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        Day 2
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../khmer-trimming/" title="Khmer Error Trimming" class="md-nav__link">
      Khmer Error Trimming
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../megahit-assembly/" title="Assembly with Megahit" class="md-nav__link">
      Assembly with Megahit
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../evaluation-and-mapping/" title="Assembly Evaluation and Quantification" class="md-nav__link">
      Assembly Evaluation and Quantification
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../annotation/" title="Assembly Annotation" class="md-nav__link">
      Assembly Annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../count_transcriptomes/" title="Count Transcriptomes" class="md-nav__link">
      Count Transcriptomes
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../filezilla/" title="Filezilla" class="md-nav__link">
      Filezilla
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../working-with-bioconda/" title="Bioconda" class="md-nav__link">
      Bioconda
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../automation-qc/" title="Workflows and Repeatability" class="md-nav__link">
      Workflows and Repeatability
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      References
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        References
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../future-installation/" title="Installing this software in the future" class="md-nav__link">
      Installing this software in the future
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../references/" title="References" class="md-nav__link">
      References
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5">
    
    <label class="md-nav__link" for="nav-5">
      For Instructors
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        For Instructors
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../for-instructors/" title="Editing this site" class="md-nav__link">
      Editing this site
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../setting-up-mkdocs/" title="Setting up MkDocs in a new repo" class="md-nav__link">
      Setting up MkDocs in a new repo
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#clean-off-old-stuff" title="Clean off old stuff" class="md-nav__link">
    Clean off old stuff
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#install-the-necessary-software-and-data" title="Install the necessary software and data" class="md-nav__link">
    Install the necessary software and data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#write-a-shell-script-to-run-rnaseq-analysis" title="Write a shell script to run RNAseq analysis" class="md-nav__link">
    Write a shell script to run RNAseq analysis
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#put-all-of-the-rnaseq-commands-into-a-single-text-file" title="Put all of the RNAseq commands into a single text file" class="md-nav__link">
    Put all of the RNAseq commands into a single text file
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#re-running-the-shell-script" title="Re-running the shell script" class="md-nav__link">
    Re-running the shell script
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#a-trick-make-it-executable" title="A trick: make it executable" class="md-nav__link">
    A trick: make it executable
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#another-good-addition-make-it-fail-nicely" title="Another good addition: make it fail nicely" class="md-nav__link">
    Another good addition: make it fail nicely
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#a-final-good-addition-make-shell-scripts-print-out-the-commands-theyre-running" title="A final good addition: make shell scripts print out the commands they're running!" class="md-nav__link">
    A final good addition: make shell scripts print out the commands they're running!
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tracking-output-of-shell-scripts" title="Tracking output of shell scripts" class="md-nav__link">
    Tracking output of shell scripts
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#a-final-note-on-shell-scripts" title="A final note on shell scripts:" class="md-nav__link">
    A final note on shell scripts:
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#snakemake-for-automation" title="snakemake for automation" class="md-nav__link">
    snakemake for automation
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#first-install-snakemake" title="First, install snakemake." class="md-nav__link">
    First, install snakemake.
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#writing-your-first-snakemake-rule" title="Writing your first snakemake rule" class="md-nav__link">
    Writing your first snakemake rule
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#challenge-add-the-explicit-list-of-inputs" title="Challenge: add the explicit list of inputs." class="md-nav__link">
    Challenge: add the explicit list of inputs.
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#challenge-write-a-rule-that-generates-the-counts-files" title="Challenge: write a rule that generates the .counts files." class="md-nav__link">
    Challenge: write a rule that generates the .counts files.
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#writing-rules-with-wildcards" title="Writing rules with wildcards" class="md-nav__link">
    Writing rules with wildcards
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#using-expand-to-work-with-lists-of-files" title="Using 'expand' to work with lists of files" class="md-nav__link">
    Using 'expand' to work with lists of files
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#challenge" title="Challenge:" class="md-nav__link">
    Challenge:
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#visualizing-workflow-diagrams" title="Visualizing workflow diagrams" class="md-nav__link">
    Visualizing workflow diagrams
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#challenge-extend-the-snakemake-rules-to-download-and-prepare-the-reference" title="Challenge: Extend the snakemake rules to download and prepare the reference" class="md-nav__link">
    Challenge: Extend the snakemake rules to download and prepare the reference
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#other-things-snakemake-can-do" title="Other things snakemake can do" class="md-nav__link">
    Other things snakemake can do
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#reproducibility-and-provenance" title="Reproducibility and provenance" class="md-nav__link">
    Reproducibility and provenance
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#answer-to-first-challenge" title="Answer to first challenge:" class="md-nav__link">
    Answer to first challenge:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#answer-to-second-challenge" title="Answer to second challenge" class="md-nav__link">
    Answer to second challenge
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#final-snakemake-file" title="Final snakemake file" class="md-nav__link">
    Final snakemake file
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                <h1 id="automation-via-shell-scripts-and-snakemake">Automation via shell scripts and snakemake</h1>
<p>Note: This lesson was written as part of <a href="http://ivory.idyll.org/dibsi/">DIBSI 2018</a> 
by C. Titus Brown. Links to the data are all provided, so you should be able to follow along. </p>
<p>Learning objectives:</p>
<ul>
<li>understand how to write shell scripts</li>
<li>understand the differences between shell scripts and workflow engines</li>
<li>develop a first snakemake workflow.</li>
</ul>
<p>We're going to reprise the RNAseq differential expression, but we're going to automate the heck out of it.</p>
<h3 id="clean-off-old-stuff">Clean off old stuff</h3>
<pre><code>cd ~/
rm -fr data rnaseq
</code></pre>

<h3 id="install-the-necessary-software-and-data">Install the necessary software and data</h3>
<p>You'll need to have conda and RStudio installed (see <a href="https://angus.readthedocs.io/en/2018/jetstream-bioconda-config.html">installation instructions</a>).</p>
<p>Then install salmon and edgeR:</p>
<pre><code>cd ~/

conda install -y salmon

curl -L -O https://raw.githubusercontent.com/ngs-docs/angus/2018/scripts/install-edgeR.R
sudo Rscript --no-save install-edgeR.R
</code></pre>

<p>You'll also need the yeast data from <a href="https://dibsi-rnaseq.readthedocs.io/en/latest/quality-trimming.html">trimming</a> again:</p>
<pre><code>cd ~/
mkdir -p data
cd data

curl -L https://osf.io/5daup/download -o ERR458493.fastq.gz
curl -L https://osf.io/8rvh5/download -o ERR458494.fastq.gz
curl -L https://osf.io/2wvn3/download -o ERR458495.fastq.gz
curl -L https://osf.io/xju4a/download -o ERR458500.fastq.gz
curl -L https://osf.io/nmqe6/download -o ERR458501.fastq.gz
curl -L https://osf.io/qfsze/download -o ERR458502.fastq.gz
</code></pre>

<p>And, finally, download the necessary analysis code into your home directory</p>
<pre><code>cd ~/
curl -L -O https://raw.githubusercontent.com/ngs-docs/angus/2018/scripts/gather-counts.py

curl -L -O https://raw.githubusercontent.com/ngs-docs/angus/2018/scripts/yeast.salmon.R
</code></pre>

<h2 id="write-a-shell-script-to-run-rnaseq-analysis">Write a shell script to run RNAseq analysis</h2>
<p>A <strong>shell script</strong> is a text file full of shell commands, that run just as if you're running them interactively at the command line.</p>
<p>Here, we'll create one that takes the commands from the RNAseq differential expression tutorial and runs 'em all at once.</p>
<h3 id="put-all-of-the-rnaseq-commands-into-a-single-text-file">Put all of the RNAseq commands into a single text file</h3>
<p>We'll script everything but the install and data download part of <a href="https://angus.readthedocs.io/en/2018/rna-seq.html">the RNAseq differential expression tutorial</a>.</p>
<p>Use RStudio to create a new file named <code>run-rnaseq.sh</code> in your home directory on the Jetstream computer, and write the following commands into it:</p>
<pre><code># make an rnaseq directory and change into it
mkdir rnaseq
cd rnaseq

# link the data in from the other directory
ln -fs ~/data/*.fastq.gz .

# download the reference transcriptome
curl -O https://downloads.yeastgenome.org/sequence/S288C_reference/orf_dna/orf_coding.fasta.gz

# index the reference transcriptome
salmon index --index yeast_orfs --type quasi --transcripts orf_coding.fasta.gz

# run salmon quant on each of the input read data sets
for i in *.fastq.gz
do
   salmon quant -i yeast_orfs --libType U -r $i -o $i.quant --seqBias --gcBias
done

# collect the counts
python2 ~/gather-counts.py

# run the edgeR script!
Rscript --no-save ~/yeast.salmon.R
</code></pre>

<p>This is now a shell script that you can use to execute all
of those commands in <em>one</em> go -- try it out! Run:</p>
<pre><code>cd ~/
bash run-rnaseq.sh
</code></pre>

<p>You should now have a directory <code>rnaseq/</code> containing a bunch of files, including the PDF outputs of the RNAseq pipeline, <code>yeast-edgeR-MA-plot.pdf</code>, <code>yeast-edgeR-MDS.pdf</code>, and <code>yeast-edgeR.csv</code>.</p>
<p>Pretty cool, eh?</p>
<h3 id="re-running-the-shell-script">Re-running the shell script</h3>
<p>Suppose you wanted to re-run the script. How would you do that?</p>
<p>Well, note that the <code>rnaseq</code> directory is created at the top of the script, and everything is executed in that directory. So if you remove the rnaseq directory like so,</p>
<pre><code>rm -fr rnaseq
</code></pre>

<p>you can then do</p>
<pre><code>bash run-rnaseq.sh
</code></pre>

<p>to just ...rerun everything.</p>
<h3 id="a-trick-make-it-executable">A trick: make it executable</h3>
<p>You can get rid of the <code>bash</code> part of the command above with
some magic:</p>
<p>Put </p>
<pre><code>#! /bin/bash
</code></pre>

<p>at the top of the file, and then run</p>
<pre><code>chmod +x ~/run-rnaseq.sh
</code></pre>

<p>at the command line.</p>
<p>You can now run</p>
<pre><code>./run-rnaseq.sh
</code></pre>

<p>instead of <code>bash run-rnaseq.sh</code>.</p>
<p>You might be thinking, ok, why is this important? Well, you can do the same with R scripts and Python scripts (but put <code>/usr/bin/env Rscript</code> or <code>/usr/bin/env python</code> at the top, instead of <code>/bin/bash</code>). This basically annotates the script with the language it's written in, so you don't have to know or remember yourself.</p>
<p>So: it's not necessary but it's a nice trick.</p>
<p>You can also always <em>force</em> a script to be run in a particular language by specifying <code>bash &lt;scriptname&gt;</code> or <code>Rscript &lt;Scriptname&gt;</code>, too.</p>
<h3 id="another-good-addition-make-it-fail-nicely">Another good addition: make it fail nicely</h3>
<p>One sort of weird aspect of shell scripts is that (by default) they keep running even if there's an error.  This is bad behavior and we should turn it off.</p>
<p>You can observe this behavior by rerunning the script above without deleting the directory <code>rnaseq/</code> - the <code>mkdir</code> command will print an error because the directory still exists, but </p>
<p>A good addition to every shell script is to make it fail on the first error. Do this by putting</p>
<pre><code>set -e
</code></pre>

<p>at the top - this tells bash to exit at the first error, rather than continuing bravely onwards.</p>
<h3 id="a-final-good-addition-make-shell-scripts-print-out-the-commands-theyre-running">A final good addition: make shell scripts print out the commands they're running!</h3>
<p>You might notice that the shell script gives you the output of the commands it's running, but doesn't tell you <em>which</em> commands it's running.</p>
<p>If you add</p>
<pre><code>set -x
</code></pre>

<p>at the top of the shell script and then re-run it,</p>
<pre><code>cd ~/
rm -fr rnaseq
./run-rnaseq.sh
</code></pre>

<p>then you'll see the full set of commands being run!</p>
<h3 id="tracking-output-of-shell-scripts">Tracking output of shell scripts</h3>
<p>you can do:</p>
<pre><code>exec &gt; output.log
exec 2&gt; error.log
</code></pre>

<p>to save all output from that point on to those files.</p>
<p>Or, you can do:</p>
<pre><code>nohup ./run-rnaseq.sh &gt; output.log 2&gt; error.log &amp;
</code></pre>

<p>and that will do the same thing, while putting it in the background.</p>
<p>You can also run 'screen' or 'tmux' (google it).</p>
<p>Torsten recommends:</p>
<pre><code>script session.log
</code></pre>

<p>While you're at it, you should look up nohup.</p>
<h3 id="a-final-note-on-shell-scripts">A final note on shell scripts:</h3>
<p><code>set -e</code> and <code>set -x</code> only work in shell scripts - they're bash commands. You need to use other approaches in Python and R.</p>
<h2 id="snakemake-for-automation">snakemake for automation</h2>
<p>Automation via shell script is wonderful, but there are a few problems here.</p>
<p>First, you have to run the entire workflow each time and it recomputes everything every time. If you're running a workflow that takes 4 days, and you change a command at the end, you'll have to manually go in and just run the stuff that depends on the changed command.</p>
<p>Second, it's very <em>explicit</em> and not very <em>generalizable</em>. If you want to run it on a different RNAseq data set, you're going to have to change a lot of commands.</p>
<p>snakemake is one of several workflow systems that help solve these problems. <a href="https://snakemake.readthedocs.io/en/stable/">(You can read the documentation here.)</a> Let's take a look!</p>
<h3 id="first-install-snakemake">First, install snakemake.</h3>
<p>Run:</p>
<pre><code>conda install -y snakemake
</code></pre>

<h2 id="writing-your-first-snakemake-rule">Writing your first snakemake rule</h2>
<p>We're going to automate the R script at the end of the shell
script in snakemake.</p>
<p>So, run the rnaseq pipeline:</p>
<pre><code>cd ~/
rm -fr rnaseq
./run-rnaseq.sh
</code></pre>

<p>Now, using RStudio, create a new text file, and put the following text in it:</p>
<pre><code>rule make_plots:
  input:
  output:
    &quot;yeast-edgeR-MA-plot.pdf&quot;,
    &quot;yeast-edgeR-MDS.pdf&quot;,
    &quot;yeast-edgeR.csv&quot;
  shell:
    &quot;Rscript --no-save ~/yeast.salmon.R&quot;
</code></pre>

<p>This rule says:
<em> to make the list of outputs (the PDFs and CSV file),
</em> you need the list of inputs (currently lbank)
* and then you run the shell command.</p>
<p>Save this with the name <code>Snakefile</code> in the <code>rnaseq/</code> directory, and run <code>snakemake make_plots</code>:</p>
<pre><code>cd ~/rnaseq
snakemake make_plots
</code></pre>

<p>you should see "Nothing to be done."</p>
<p>That's because the PDFs and CSV file already exist!</p>
<p>Let's fix that:</p>
<pre><code>rm yeast-*.pdf yeast-*.csv
</code></pre>

<p>and now, when you run <code>snakemake make_plots</code>, you should see the R script being run. Yay w00t! Then if you run <code>snakemake make_plots</code> again, you will see that it doesn't need to do anything - all the files are "up to date".</p>
<h4 id="challenge-add-the-explicit-list-of-inputs">Challenge: add the explicit list of inputs.</h4>
<p>What does the <code>make_plots</code> rule take as input? List them!</p>
<p>Hint: what does the <code>yeast.salmon.R</code> script take in?</p>
<hr />
<p>Once you're done, you can also update the timestamp on the counts files and it will regenerate the PDFs --</p>
<pre><code>touch *.counts
snakemake make_plots
</code></pre>

<p>because it knows that the files the rule depends on have changed!</p>
<h4 id="challenge-write-a-rule-that-generates-the-counts-files">Challenge: write a rule that generates the .counts files.</h4>
<p>Looking at the shell script, write a new Snakemake rule that generates the counts files as outputs.</p>
<p>You can start by adding a new blank rule:</p>
<pre><code>rule make_counts:
  input:
  output:
  shell:
    &quot;&quot;
</code></pre>

<p>and filling in from there. Note that rule order does not matter in the Snakefile!</p>
<p>Hints:</p>
<ul>
<li>you should list each .counts file separately.</li>
</ul>
<p>To test the rule, do</p>
<pre><code>rm *.counts yeast-*.pdf
</code></pre>

<p>and it will regenerate everything needed to build the PDF!</p>
<h3 id="writing-rules-with-wildcards">Writing rules with wildcards</h3>
<p>Let's suppose we want to write a rule that takes a FASTQ file (say, <code>ERR458501.fastq.gz</code>) and runs salmon on it to generate the directory <code>ERR458501.fastq.gz.quant</code>. How would we do that?</p>
<p>We could write the rule as follows:</p>
<pre><code>rule generate_quant_dir_ERR458501:
    input:
        &quot;ERR458501.fastq.gz&quot;
    output:
        directory(&quot;ERR458501.fastq.gz.quant&quot;)
    shell:
        &quot;salmon quant -i yeast_orfs --libType U -r ERR458501.fastq.gz -o ERR458501.fastq.gz.quant --seqBias --gcBias&quot;
</code></pre>

<p>and that will work just fine -- you can run,</p>
<pre><code>rm -fr ERR458501.fastq.gz.quant
snakemake ERR458501.fastq.gz.quant
</code></pre>

<p>and it will happily run salmon for you.</p>
<p>But there's a lot of repeated stuff in there. Can we do cool computer things to fix that? Yes! We can use wildcards!</p>
<p>Try replacing <code>ERR458501</code> with <code>{accession}</code> - this makes it into a <em>wildcard</em> that snakemake can automatically fill in:</p>
<pre><code>rule generate_quant_dir:
  input:
    &quot;{accession}.fastq.gz&quot;
  output:
    directory(&quot;{accession}.fastq.gz.quant&quot;)
  shell:
    &quot;salmon quant -i yeast_orfs --libType U -r {wildcards.accession}.fastq.gz -o {wildcards.accession}.fastq.gz.quant --seqBias --gcBias&quot;
</code></pre>

<p>Now, you can run</p>
<pre><code>snakemake ERR458501.fastq.gz.quant
</code></pre>

<p>and it will work - but you can <em>also</em> run</p>
<pre><code>snakemake ERR458493.fastq.gz.quant
</code></pre>

<p>and it will do the right thing there, as well.</p>
<h3 id="using-expand-to-work-with-lists-of-files">Using 'expand' to work with lists of files</h3>
<p>The challenge to write a rule that generates the .counts files should have left you with a list of six input files (<code>.quant</code> directories), and <code>make_plots</code> has a very similar of six input files (<code>.quant.counts</code>).</p>
<p>Can we make this more succinct? Yep. You can have a list that you <strong>expand</strong> in various ways.  First, define the list of files at the top of the Snakefile:</p>
<pre><code>input_files=[&quot;ERR458493.fastq.gz&quot;,
  &quot;ERR458494.fastq.gz&quot;,
  &quot;ERR458495.fastq.gz&quot;,
  &quot;ERR458500.fastq.gz&quot;,
  &quot;ERR458501.fastq.gz&quot;,
  &quot;ERR458502.fastq.gz&quot;]
</code></pre>

<p>now, modify the <code>make_plots</code> rule to have the input be:</p>
<pre><code>rule make_plots:
   input:
      expand(&quot;{input_files}.quant.counts&quot;, input_files=input_files)
</code></pre>

<h4 id="challenge">Challenge:</h4>
<p>Update the other rules that have lists of six files using <code>expand</code>.</p>
<h3 id="visualizing-workflow-diagrams">Visualizing workflow diagrams</h3>
<p>Try running:</p>
<pre><code>snakemake --dag yeast-edgeR.csv |  dot -Tpng &gt; dag.png
</code></pre>

<h3 id="challenge-extend-the-snakemake-rules-to-download-and-prepare-the-reference">Challenge: Extend the snakemake rules to download and prepare the reference</h3>
<p>'nuff said'</p>
<h3 id="other-things-snakemake-can-do">Other things snakemake can do</h3>
<p>You can specify what programs your pipeline depends in on your snakemake setup! snakemake interfaces well with conda (in fact, the authors of snakemake are also leads on the bioconda project) so you can pretty much just list your packages.</p>
<h2 id="reproducibility-and-provenance">Reproducibility and provenance</h2>
<p>If you write all your workflows like this, all you need to do is give readers your data files, your Snakefile, and your list of software... and they can reproduce everything you did!</p>
<h3 id="answer-to-first-challenge">Answer to first challenge:</h3>
<pre><code>rule make_plots:
  input:
    &quot;ERR458493.fastq.gz.quant.counts&quot;,
    &quot;ERR458500.fastq.gz.quant.counts&quot;,
    &quot;ERR458494.fastq.gz.quant.counts&quot;, 
    &quot;ERR458501.fastq.gz.quant.counts&quot;,
    &quot;ERR458495.fastq.gz.quant.counts&quot;,
    &quot;ERR458502.fastq.gz.quant.counts&quot;
  output:
    &quot;yeast-edgeR-MA-plot.pdf&quot;,
    &quot;yeast-edgeR-MDS.pdf&quot;,
    &quot;yeast-edgeR.csv&quot;
  shell:
    &quot;Rscript --no-save ~/yeast.salmon.R&quot;
</code></pre>

<h3 id="answer-to-second-challenge">Answer to second challenge</h3>
<pre><code># make the .counts files
rule make_counts:
  input:
    &quot;ERR458493.fastq.gz.quant&quot;,
    &quot;ERR458500.fastq.gz.quant&quot;,
    &quot;ERR458494.fastq.gz.quant&quot;, 
    &quot;ERR458501.fastq.gz.quant&quot;,
    &quot;ERR458495.fastq.gz.quant&quot;,
    &quot;ERR458502.fastq.gz.quant&quot;
  output:
    &quot;ERR458493.fastq.gz.quant.counts&quot;,
    &quot;ERR458500.fastq.gz.quant.counts&quot;,
    &quot;ERR458494.fastq.gz.quant.counts&quot;, 
    &quot;ERR458501.fastq.gz.quant.counts&quot;,
    &quot;ERR458495.fastq.gz.quant.counts&quot;,
    &quot;ERR458502.fastq.gz.quant.counts&quot;
  shell:
    &quot;python2 ~/gather-counts.py&quot;
</code></pre>

<h3 id="final-snakemake-file">Final snakemake file</h3>
<pre><code>input_files=[&quot;ERR458493.fastq.gz&quot;,
  &quot;ERR458494.fastq.gz&quot;,
  &quot;ERR458495.fastq.gz&quot;,
  &quot;ERR458500.fastq.gz&quot;,
  &quot;ERR458501.fastq.gz&quot;,
  &quot;ERR458502.fastq.gz&quot;]

rule make_plots:
  input:
    expand(&quot;{input_files}.quant.counts&quot;, input_files=input_files)
  output:
    &quot;yeast-edgeR-MA-plot.pdf&quot;,
    &quot;yeast-edgeR-MDS.pdf&quot;,
    &quot;yeast-edgeR.csv&quot;
  shell:
    &quot;Rscript --no-save ~/yeast.salmon.R&quot;


# make the .counts files
rule make_counts:
  input:
    expand(&quot;{input_files}.quant&quot;, input_files=input_files)
  output:
    expand(&quot;{input_files}.quant.counts&quot;, input_files=input_files)
  shell:
    &quot;python2 ~/gather-counts.py&quot;

rule generate_quant_dir:
  input:
    &quot;{accession}.fastq.gz&quot;
  output:
    &quot;{accession}.fastq.gz.quant&quot;
  shell:
    &quot;salmon quant -i yeast_orfs --libType U -r {wildcards.accession}.fastq.gz -o {wildcards.accession}.fastq.gz.quant --seqBias --gcBias&quot;
</code></pre>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright &copy; 2018 <a href="http://ivory.idyll.org/lab/">Lab for Data Intensive Biology</a> at UC Davis
          </div>
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/application.e72fd936.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:".."}})</script>
      
        <script src="../search/main.js"></script>
      
    
    
      
    
  </body>
</html>