Environmental Metatranscriptomics, Nov 2018 
======

These are the online materials for the environmental metatranscriptomics workshop to be run at [CICESE](https://www.cicese.edu.mx/).

This workshop runs under a [Code of Conduct](code-of-conduct.md). Please respect it and be excellent to each other!


## Day 1: Nov 8, 2018

**Morning:**
  
  - Introduction and Metatranscriptomics overview 
  - [Bienvenidos!](welcome.md)
  - [Logging into the cluster](cicese-cluster.md) 
  - [Setting up the software](setting-up-tara-environment.md)
  - [Intro to TARA Oceans sample data](tara-sample-data.md)
  - [Short read quality control](short-read-quality-control.md)

Lunch 

**Afternoon:**
  
  
  - [Taxonomic classification with sourmash gather](sourmash-taxonomic-classification.md)
  - [Sample Comparison](sample-comparison.md)
  - [Error Trimming](khmer-trimming.md)
  - [Metatrancriptome Assembly](megahit-assembly.md)
  - [Counting transcriptomes](count_transcriptomes.md)
  


## Day 2: Nov 9, 2018

**Morning:** 

Working with Reads
  
  - [Assessing assembly quality](evaluation.md)
  - [Read mapping and quantification](read-mapping.md)
  - if time: [Annotating your assembly](annotation.md)
  
  - Metatranscriptomic workflows and considerations

Lunch 

**Afternoon:** 

  - [Bioconda for software installation](working-with-bioconda.md)
  - Snakemake, bioconda etc? 
  - CLUST for co-expressed gene clusters!? [CLUST paper](https://www.biorxiv.org/content/early/2018/02/13/221309)
  - [Putting it all together with Anviâ€™o](anvio.md)
  - A brief discussion of workflows & repeatability




#### Additional Resources:  

  - SEQ Answers
  - Biostars
  - Data Carpentry
  - DIB Summer Institute


#### Suggested readings

  - [A global ocean atlas of eukaryotic genes](https://www.nature.com/articles/s41467-017-02342-1) - This is the paper from which we pulled all of our sample data.

