Environmental Metatranscriptomics, Nov 2018 
======

These are the online materials for the environmental metatranscriptomics workshop run at [CICESE](https://www.cicese.edu.mx/).

This workshop runs under a [Code of Conduct](code-of-conduct.md). Please respect it and be excellent to each other!

*Currently just a condensed version of 2017 metagenomics workshop at CICESE*

Day 1: Nov X, 2018

Morning:
 - Logging into cloud compute (aws?)
 - Working on the command line
 - Evaluating short-read quality

Lunch 

Afternoon:
 - Metatranscriptomic workflows and considerations
 - Assembling your short read data set with MEGAHIT
 - Evaluating your assemblies

Day 2: Nov X, 2018

Morning:
 - Mapping short reads to the assembly time permitting
 - Sourmash compare
 - Annotating your assembly with Prokka
 - Binning genomes out of your metagenome

Lunch 

Afternoon: 
 - Quantifying abundance across samples with Salmon
 - Taxonomic classification with sourmash gather
 - [Putting it all together with Anviâ€™o](anvio.md))
 - A brief discussion of workflows & repeatability


Readings:
 - Read the Critical Assessment of Metagenome Interpretation (CAMI) Paper;
 - Read Hu et al. 2016. This is the paper from which we pulled all of our sample data.

Additional Resources:

 - SEQ Answers
 - Biostars
 - Data Carpentry
 - DIB Summer Institute


