Environmental Metatranscriptomics, Nov 2018 
======

These are the online materials for the environmental metatranscriptomics workshop to be run at [CICESE](https://www.cicese.edu.mx/).

This workshop runs under a [Code of Conduct](code-of-conduct.md). Please respect it and be excellent to each other!

_Tentative Schedule_

Day 1: Nov 8, 2018

Morning:  
  
  - Introduction and Metatranscriptomics overview 
  - [Bienvenidos!](welcome.md)
  - [Logging into the cluster](cicese-cluster.md) 
  - [Bioconda for software installation](working-with-bioconda.md)
  - [Short read quality control](short-read-quality-control.md)
  - **If time:** [Metatrancriptome Assembly](metatranscriptome-assembly.md)

Lunch 

Afternoon: 

  - Metatranscriptomic workflows and considerations
  - [Intro to TARA Oceans sample data](tara-sample-data.md)
  - [Assessing assembly quality](metatranscriptome-evaluation.md)
  - **If time:** [Annotating your assembly](metatranscriptome-annotation.md)


Day 2: Nov 9, 2018

Morning:  
  
  - [Counting transcriptomes](count_transcriptomes.md)
  - [Taxonomic classification with sourmash gather](sourmash-taxonomic-classification.md)
  - [Read mapping and quantification](read-mapping.md)

Lunch 

Afternoon:  

  - Differential Expression?
  - CLUST for co-expressed gene clusters!? [CLUST paper](https://www.biorxiv.org/content/early/2018/02/13/221309)
  - [Putting it all together with Anviâ€™o](anvio.md)
  - A brief discussion of workflows & repeatability




#### Additional Resources:  

  - SEQ Answers
  - Biostars
  - Data Carpentry
  - DIB Summer Institute


#### CICESE 2017 readings:  

  - Read the Critical Assessment of Metagenome Interpretation (CAMI) Paper;
  - Read Hu et al. 2016. This is the paper from which we pulled all of our sample data.

