# Some other useful references

We've covered some tools during this workshop that we often use in our workflows, 
but there are others that may suit your purposes better. We have listed some of 
them below, as well as some useful observations we have found when using the 
tools.

## Finding Data
+ [Sequence Read Archive](https://www.ncbi.nlm.nih.gov/sra/docs/)
+ [European Nucleotide Archive](https://www.ebi.ac.uk/ena)
+ [GenBank](https://www.ncbi.nlm.nih.gov/genbank/)
+ [NCBI Taxonomy](https://www.ncbi.nlm.nih.gov/taxonomy)

## Quality Trimming and Evalution
+ [Trimmomatic](http://www.usadellab.org/cms/?page=trimmomatic)
+ [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
+ [MultiQC](https://multiqc.info/)
+ [CutAdapt](https://cutadapt.readthedocs.io/en/stable/guide.html) -- Very good at removing polyA tails

## Transcriptome and Metatranscriptome assembly

+ [MEGAHIT](https://www.ncbi.nlm.nih.gov/pubmed/25609793)
+ [Trinity](https://github.com/trinityrnaseq/trinityrnaseq/wiki)
+ [PLASS](https://www.biorxiv.org/content/early/2018/08/07/386110)

## Assembly Evaluation

+ [BUSCO](https://busco.ezlab.org/)
+ [Transrate](http://hibberdlab.com/transrate/)

## Read Mapping and Quantification

+ [Salmon](https://www.nature.com/articles/nmeth.4197)
+ [STAR](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4631051/)
+ [BWA](http://bio-bwa.sourceforge.net/bwa.shtml)
+ [Bowtie2](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3322381/)

## Amplicon Sequencing

+ [QIIME](http://qiime.org/)



